
<!DOCTYPE html>
<!--[if lt IE 7]><html class="lt-ie9 lt-ie8 lt-ie7" lang="en" dir="ltr"><![endif]-->
<!--[if IE 7]><html class="lt-ie9 lt-ie8" lang="en" dir="ltr"><![endif]-->
<!--[if IE 8]><html class="lt-ie9" lang="en" dir="ltr"><![endif]-->
<!--[if gt IE 8]><!--><html lang="en" dir="ltr" prefix="content: http://purl.org/rss/1.0/modules/content/ dc: http://purl.org/dc/terms/ foaf: http://xmlns.com/foaf/0.1/ og: http://ogp.me/ns# rdfs: http://www.w3.org/2000/01/rdf-schema# sioc: http://rdfs.org/sioc/ns# sioct: http://rdfs.org/sioc/types# skos: http://www.w3.org/2004/02/skos/core# xsd: http://www.w3.org/2001/XMLSchema#"><!--<![endif]-->
<head>
<meta charset="utf-8" />
<meta name="Generator" content="Drupal 7 (http://drupal.org)" />
<link rel="canonical" href="biovenn-r-and-python-package-comparison-and-visualization-biological-lists-using-area.html" />
<link rel="shortlink" href="../node/679.html" />
<link rel="shortcut icon" href="../sites/default/files/favicon-96x96.png" type="image/png" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta name="MobileOptimized" content="width" />
<meta name="HandheldFriendly" content="true" />
<meta name="apple-mobile-web-app-capable" content="yes" />
<title>BioVenn - an R and Python package for the comparison and visualization of biological lists using area-proportional Venn diagrams | Data Science</title>
<style type="text/css" media="all">
@import url("https://datasciencehub.net/modules/system/system.base.css?t4kojz");
@import url("https://datasciencehub.net/modules/system/system.menus.css?t4kojz");
@import url("https://datasciencehub.net/modules/system/system.messages.css?t4kojz");
@import url("https://datasciencehub.net/modules/system/system.theme.css?t4kojz");
</style>
<style type="text/css" media="all">
@import url("https://datasciencehub.net/misc/ui/jquery.ui.core.css?t4kojz");
@import url("https://datasciencehub.net/misc/ui/jquery.ui.theme.css?t4kojz");
</style>
<style type="text/css" media="all">
@import url("https://datasciencehub.net/modules/comment/comment.css?t4kojz");
@import url("https://datasciencehub.net/modules/field/theme/field.css?t4kojz");
@import url("../sites/all/modules/logintoboggan/logintoboggan.css%3Ft4kojz.css");
@import url("https://datasciencehub.net/modules/node/node.css?t4kojz");
@import url("https://datasciencehub.net/modules/search/search.css?t4kojz");
@import url("https://datasciencehub.net/modules/user/user.css?t4kojz");
@import url("../sites/all/modules/views/css/views.css%3Ft4kojz.css");
</style>
<style type="text/css" media="all">
@import url("../sites/all/modules/ctools/css/ctools.css%3Ft4kojz.css");
@import url("../sites/all/modules/ctools/css/modal.css%3Ft4kojz.css");
@import url("../sites/default/modules/modal_forms/css/modal_forms_popup.css%3Ft4kojz.css");
@import url("../sites/all/modules/responsive_menus/styles/responsive_menus_simple/css/responsive_menus_simple.css%3Ft4kojz.css");
@import url("../sites/all/modules/date/date_api/date.css%3Ft4kojz.css");
</style>
<style type="text/css" media="screen">
@import url("../sites/all/themes/adaptivetheme/at_core/css/at.settings.style.headings.css%3Ft4kojz.css");
@import url("../sites/all/themes/adaptivetheme/at_core/css/at.settings.style.image.css%3Ft4kojz.css");
@import url("../sites/all/themes/adaptivetheme/at_core/css/at.layout.css%3Ft4kojz.css");
</style>
<style type="text/css" media="all">
@import url("../sites/all/themes/corolla/css/html-elements.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/forms.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/tables.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/page.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/articles.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/comments.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/fields.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/blocks.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/navigation.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/corolla.settings.style.css%3Ft4kojz.css");
@import url("../sites/default/files/color/corolla-80fb2671/colors.css%3Ft4kojz.css");
</style>
<style type="text/css" media="print">
@import url("../sites/all/themes/corolla/css/print.css%3Ft4kojz.css");
</style>
<link type="text/css" rel="stylesheet" href="../sites/default/files/adaptivetheme/corolla_files/corolla.responsive.layout.css%3Ft4kojz.css" media="only screen" />
<style type="text/css" media="screen">
@import url("../sites/default/files/adaptivetheme/corolla_files/corolla.fonts.css%3Ft4kojz.css");
</style>
<link type="text/css" rel="stylesheet" href="../sites/all/themes/corolla/css/responsive.custom.css%3Ft4kojz.css" media="only screen" />
<link type="text/css" rel="stylesheet" href="../sites/all/themes/corolla/css/responsive.smartphone.portrait.css%3Ft4kojz.css" media="only screen and (max-width:320px)" />
<link type="text/css" rel="stylesheet" href="../sites/all/themes/corolla/css/responsive.smartphone.landscape.css%3Ft4kojz.css" media="only screen and (min-width:321px) and (max-width:480px)" />
<link type="text/css" rel="stylesheet" href="../sites/all/themes/corolla/css/responsive.tablet.portrait.css%3Ft4kojz.css" media="only screen and (min-width:481px) and (max-width:768px)" />
<link type="text/css" rel="stylesheet" href="../sites/all/themes/corolla/css/responsive.tablet.landscape.css%3Ft4kojz.css" media="only screen and (min-width:769px) and (max-width:1024px)" />
<link type="text/css" rel="stylesheet" href="../sites/all/themes/corolla/css/responsive.desktop.css%3Ft4kojz.css" media="only screen and (min-width:1025px)" />

<!--[if lt IE 9]>
<style type="text/css" media="screen">
@import url("https://datasciencehub.net/sites/default/files/adaptivetheme/corolla_files/corolla.lt-ie9.layout.css?t4kojz");
</style>
<![endif]-->

<!--[if lte IE 9]>
<style type="text/css" media="screen">
@import url("https://datasciencehub.net/sites/all/themes/corolla/css/ie-lte-9.css?t4kojz");
</style>
<![endif]-->
<script type="text/javascript" src="https://datasciencehub.net/misc/jquery.js?v=1.4.4"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/jquery-extend-3.4.0.js?v=1.4.4"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/jquery-html-prefilter-3.5.0-backport.js?v=1.4.4"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/jquery.once.js?v=1.2"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/drupal.js?t4kojz"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/ui/jquery.ui.core.min.js?v=1.8.7"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/jquery.cookie.js?v=1.0"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/jquery.form.js?v=2.52"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/form-single-submit.js?v=7.98"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/form.js?v=7.98"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/ajax.js?v=7.98"></script>
<script type="text/javascript" src="../sites/default/modules/jquery_update/js/jquery_update.js%3Fv=0.0.1"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/progress.js?v=7.98"></script>
<script type="text/javascript" src="../sites/all/modules/ctools/js/modal.js%3Ft4kojz"></script>
<script type="text/javascript" src="../sites/default/modules/modal_forms/js/modal_forms_popup.js%3Ft4kojz"></script>
<script type="text/javascript" src="../sites/all/modules/responsive_menus/styles/responsive_menus_simple/js/responsive_menus_simple.js%3Ft4kojz"></script>
<script type="text/javascript" src="../sites/all/modules/google_analytics/googleanalytics.js%3Ft4kojz"></script>
<script type="text/javascript">
<!--//--><![CDATA[//><!--
(function(i,s,o,g,r,a,m){i["GoogleAnalyticsObject"]=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)})(window,document,"script","https://www.google-analytics.com/analytics.js","ga");ga("create", "UA-107867597-1", {"cookieDomain":"auto"});ga("set", "anonymizeIp", true);ga("send", "pageview");
//--><!]]>
</script>
<script type="text/javascript" src="../sites/all/modules/field_group/field_group.js%3Ft4kojz"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/tableheader.js?t4kojz"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/collapse.js?v=7.98"></script>
<script type="text/javascript">
<!--//--><![CDATA[//><!--
jQuery.extend(Drupal.settings, {"basePath":"\/","pathPrefix":"","setHasJsCookie":0,"ajaxPageState":{"theme":"corolla","theme_token":"AB4rZRX_dEhfgbn40de-Hp9vYHqcaN7cw2vZpyzBaOQ","jquery_version":"default","js":{"misc\/jquery.js":1,"misc\/jquery-extend-3.4.0.js":1,"misc\/jquery-html-prefilter-3.5.0-backport.js":1,"misc\/jquery.once.js":1,"misc\/drupal.js":1,"misc\/ui\/jquery.ui.core.min.js":1,"misc\/jquery.cookie.js":1,"misc\/jquery.form.js":1,"misc\/form-single-submit.js":1,"misc\/form.js":1,"misc\/ajax.js":1,"sites\/default\/modules\/jquery_update\/js\/jquery_update.js":1,"misc\/progress.js":1,"sites\/all\/modules\/ctools\/js\/modal.js":1,"sites\/default\/modules\/modal_forms\/js\/modal_forms_popup.js":1,"sites\/all\/modules\/responsive_menus\/styles\/responsive_menus_simple\/js\/responsive_menus_simple.js":1,"sites\/all\/modules\/google_analytics\/googleanalytics.js":1,"0":1,"sites\/all\/modules\/field_group\/field_group.js":1,"misc\/tableheader.js":1,"misc\/collapse.js":1},"css":{"modules\/system\/system.base.css":1,"modules\/system\/system.menus.css":1,"modules\/system\/system.messages.css":1,"modules\/system\/system.theme.css":1,"misc\/ui\/jquery.ui.core.css":1,"misc\/ui\/jquery.ui.theme.css":1,"modules\/comment\/comment.css":1,"modules\/field\/theme\/field.css":1,"sites\/all\/modules\/logintoboggan\/logintoboggan.css":1,"modules\/node\/node.css":1,"modules\/search\/search.css":1,"modules\/user\/user.css":1,"sites\/all\/modules\/views\/css\/views.css":1,"sites\/all\/modules\/ctools\/css\/ctools.css":1,"sites\/all\/modules\/ctools\/css\/modal.css":1,"sites\/default\/modules\/modal_forms\/css\/modal_forms_popup.css":1,"sites\/all\/modules\/responsive_menus\/styles\/responsive_menus_simple\/css\/responsive_menus_simple.css":1,"sites\/all\/modules\/date\/date_api\/date.css":1,"sites\/all\/themes\/adaptivetheme\/at_core\/css\/at.settings.style.headings.css":1,"sites\/all\/themes\/adaptivetheme\/at_core\/css\/at.settings.style.image.css":1,"sites\/all\/themes\/adaptivetheme\/at_core\/css\/at.layout.css":1,"sites\/all\/themes\/corolla\/css\/html-elements.css":1,"sites\/all\/themes\/corolla\/css\/forms.css":1,"sites\/all\/themes\/corolla\/css\/tables.css":1,"sites\/all\/themes\/corolla\/css\/page.css":1,"sites\/all\/themes\/corolla\/css\/articles.css":1,"sites\/all\/themes\/corolla\/css\/comments.css":1,"sites\/all\/themes\/corolla\/css\/fields.css":1,"sites\/all\/themes\/corolla\/css\/blocks.css":1,"sites\/all\/themes\/corolla\/css\/navigation.css":1,"sites\/all\/themes\/corolla\/css\/fonts.css":1,"sites\/all\/themes\/corolla\/css\/corolla.settings.style.css":1,"sites\/all\/themes\/corolla\/color\/colors.css":1,"sites\/all\/themes\/corolla\/css\/print.css":1,"public:\/\/adaptivetheme\/corolla_files\/corolla.responsive.layout.css":1,"public:\/\/adaptivetheme\/corolla_files\/corolla.fonts.css":1,"sites\/all\/themes\/corolla\/css\/responsive.custom.css":1,"sites\/all\/themes\/corolla\/css\/responsive.smartphone.portrait.css":1,"sites\/all\/themes\/corolla\/css\/responsive.smartphone.landscape.css":1,"sites\/all\/themes\/corolla\/css\/responsive.tablet.portrait.css":1,"sites\/all\/themes\/corolla\/css\/responsive.tablet.landscape.css":1,"sites\/all\/themes\/corolla\/css\/responsive.desktop.css":1,"public:\/\/adaptivetheme\/corolla_files\/corolla.lt-ie9.layout.css":1,"sites\/all\/themes\/corolla\/css\/ie-lte-9.css":1}},"CToolsModal":{"loadingText":"Loading...","closeText":"Close Window","closeImage":"\u003Cimg typeof=\u0022foaf:Image\u0022 class=\u0022image-style-none\u0022 src=\u0022https:\/\/datasciencehub.net\/sites\/all\/modules\/ctools\/images\/icon-close-window.png\u0022 alt=\u0022Close window\u0022 title=\u0022Close window\u0022 \/\u003E","throbber":"\u003Cimg typeof=\u0022foaf:Image\u0022 class=\u0022image-style-none\u0022 src=\u0022https:\/\/datasciencehub.net\/sites\/all\/modules\/ctools\/images\/throbber.gif\u0022 alt=\u0022Loading\u0022 title=\u0022Loading...\u0022 \/\u003E"},"modal-popup-small":{"modalSize":{"type":"fixed","width":500,"height":150},"modalOptions":{"opacity":0.85,"background":"#000"},"animation":"fadeIn","modalTheme":"ModalFormsPopup","throbber":"\u003Cimg typeof=\u0022foaf:Image\u0022 class=\u0022image-style-none\u0022 src=\u0022https:\/\/datasciencehub.net\/sites\/default\/modules\/modal_forms\/images\/loading_animation.gif\u0022 alt=\u0022Loading...\u0022 title=\u0022Loading\u0022 \/\u003E","closeText":"Close"},"modal-popup-medium":{"modalSize":{"type":"fixed","width":550,"height":450},"modalOptions":{"opacity":0.85,"background":"#000"},"animation":"fadeIn","modalTheme":"ModalFormsPopup","throbber":"\u003Cimg typeof=\u0022foaf:Image\u0022 class=\u0022image-style-none\u0022 src=\u0022https:\/\/datasciencehub.net\/sites\/default\/modules\/modal_forms\/images\/loading_animation.gif\u0022 alt=\u0022Loading...\u0022 title=\u0022Loading\u0022 \/\u003E","closeText":"Close"},"modal-popup-large":{"modalSize":{"type":"scale","width":0.8,"height":0.8},"modalOptions":{"opacity":0.85,"background":"#000"},"animation":"fadeIn","modalTheme":"ModalFormsPopup","throbber":"\u003Cimg typeof=\u0022foaf:Image\u0022 class=\u0022image-style-none\u0022 src=\u0022https:\/\/datasciencehub.net\/sites\/default\/modules\/modal_forms\/images\/loading_animation.gif\u0022 alt=\u0022Loading...\u0022 title=\u0022Loading\u0022 \/\u003E","closeText":"Close"},"responsive_menus":[{"toggler_text":"\u2630 Menu","selectors":["#block-system-main-menu"],"media_size":"935","media_unit":"px","absolute":"1","remove_attributes":"1","responsive_menus_style":"responsive_menus_simple"}],"googleanalytics":{"trackOutbound":1,"trackMailto":1,"trackDownload":1,"trackDownloadExtensions":"7z|aac|arc|arj|asf|asx|avi|bin|csv|doc(x|m)?|dot(x|m)?|exe|flv|gif|gz|gzip|hqx|jar|jpe?g|js|mp(2|3|4|e?g)|mov(ie)?|msi|msp|pdf|phps|png|ppt(x|m)?|pot(x|m)?|pps(x|m)?|ppam|sld(x|m)?|thmx|qtm?|ra(m|r)?|sea|sit|tar|tgz|torrent|txt|wav|wma|wmv|wpd|xls(x|m|b)?|xlt(x|m)|xlam|xml|z|zip"},"urlIsAjaxTrusted":{"\/search\/node":true,"\/paper\/biovenn-r-and-python-package-comparison-and-visualization-biological-lists-using-area":true},"field_group":{"fieldset":"full"},"ctools-sample-style":{"modalSize":{"type":"fixed","width":1098,"height":500,"addWidth":20,"addHeight":15},"modalOptions":{"opacity":0.5,"background-color":"#000"},"animation":"fadeIn","modalTheme":"CToolsSampleModal","throbber":"\u003Cimg typeof=\u0022foaf:Image\u0022 class=\u0022image-style-none\u0022 src=\u0022https:\/\/datasciencehub.net\/sites\/all\/modules\/ctools\/ctools_ajax_sample\/images\/ajax-loader.gif\u0022 alt=\u0022Loading...\u0022 title=\u0022Loading\u0022 \/\u003E"},"adaptivetheme":{"corolla":{"layout_settings":{"bigscreen":"three-col-grail","tablet_landscape":"three-col-grail","tablet_portrait":"one-col-vert","smalltouch_landscape":"one-col-vert","smalltouch_portrait":"one-col-stack"},"media_query_settings":{"bigscreen":"only screen and (min-width:1025px)","tablet_landscape":"only screen and (min-width:769px) and (max-width:1024px)","tablet_portrait":"only screen and (min-width:481px) and (max-width:768px)","smalltouch_landscape":"only screen and (min-width:321px) and (max-width:480px)","smalltouch_portrait":"only screen and (max-width:320px)"}}}});
//--><!]]>
</script>
<!--[if lt IE 9]>
<script src="https://datasciencehub.net/sites/all/themes/adaptivetheme/at_core/scripts/html5.js?t4kojz"></script>
<![endif]-->
</head>
<body class="html not-front not-logged-in no-sidebars page-node page-node- page-node-679 node-type-submit-paper site-name-hidden atr-7.x-3.x atv-7.x-3.1 lang-en site-name-data-science section-paper color-scheme-custom corolla bs-n bb-n mb-dd rc-6 rct-6">
  <div id="skip-link" class="nocontent">
    <a href="biovenn-r-and-python-package-comparison-and-visualization-biological-lists-using-area.html#main-content" class="element-invisible element-focusable">Skip to main content</a>
  </div>
    <div id="page-wrapper">
  <div id="page" class="page ssc-n ssw-n ssa-l sss-n btc-n btw-b bta-l bts-n ntc-n ntw-b nta-l nts-n ctc-n ctw-b cta-l cts-n ptc-n ptw-b pta-l pts-n">

          <div id="menu-bar-wrapper">
        <div class="container clearfix">
          <div id="menu-bar" class="nav clearfix"><nav id="block-system-main-menu" class="block block-system block-menu menu-wrapper menu-bar-wrapper clearfix odd first block-count-1 block-region-menu-bar block-main-menu"  role="navigation">  
      <h2 class="element-invisible block-title">Main menu</h2>
  
  <ul class="menu clearfix"><li class="first leaf menu-depth-1 menu-item-757"><a href="../content/about-data-science.html">About</a></li><li class="leaf menu-depth-1 menu-item-758"><a href="../content/editors.html">Editorial Board</a></li><li class="leaf menu-depth-1 menu-item-1559"><a href="../content/submit-manuscript.html">Submit Manuscript</a></li><li class="leaf menu-depth-1 menu-item-1612"><a href="../content/calls.html">Calls</a></li><li class="leaf menu-depth-1 menu-item-759"><a href="../content/guidelines-authors.html">For Authors</a></li><li class="leaf menu-depth-1 menu-item-760"><a href="../content/guidelines-reviewers.html">For Reviewers</a></li><li class="leaf menu-depth-1 menu-item-582"><a href="../underreview.html" title="Under Review">Under Review</a></li><li class="leaf menu-depth-1 menu-item-583"><a href="../reviewed.html" title="">Reviewed</a></li><li class="last leaf menu-depth-1 menu-item-1611"><a href="../ReviewedAccepted.html" title="">Accepted</a></li></ul>
  </nav><div id="block-orcid-login-orcid-login" class="block block-orcid-login no-title menu-wrapper menu-bar-wrapper clearfix even last block-count-2 block-region-menu-bar block-orcid-login" >  
  
  <p style="background-color: #003b3a; float:right;"><a style="vertical-align: middle; color: white; padding: 10px; display: inline-block; text-decoration: none;" href="https://orcid.org/oauth/authorize?client_id=APP-223SGGDDO35LKOT2&amp;response_type=code&amp;scope=%2Fauthenticate&amp;redirect_uri=https%3A%2F%2Fdatasciencehub.net%2F&amp;state=eyJyZWRpcmVjdCI6InBhcGVyXC9iaW92ZW5uLXItYW5kLXB5dGhvbi1wYWNrYWdlLWNvbXBhcmlzb24tYW5kLXZpc3VhbGl6YXRpb24tYmlvbG9naWNhbC1saXN0cy11c2luZy1hcmVhIn0%3D"><img alt="ORCID logo" src="../sites/default/files/orcid_24x24.png" style="vertical-align: middle;"/> Log In with ORCID</a></p>
  </div></div>        </div>
      </div>
    
    <div id="header-wrapper">
      <div class="container clearfix">
	        <header class="clearfix with-logo" role="banner">

                      <div id="branding" class="branding-elements clearfix">

                              <div id="logo">
                  <a href="../index.html"><img class="site-logo" typeof="foaf:Image" src="../sites/default/files/DS_LOGO_vertical.png" alt="Data Science" /></a>                </div>
              
                              <div class="element-invisible h-group" id="name-and-slogan">

                                      <h1 class="element-invisible" id="site-name"><a href="../index.html" title="Home page">Data Science</a></h1>
                  
                  
                </div>
              
            </div>
          
          
        </header>

      </div>
    </div>

          <div id="top-panels-wrapper">
        <div class="container clearfix">
          <!-- Three column 3x33 Gpanel -->
          <div class="at-panel gpanel panel-display three-3x33 clearfix">
                                                          </div>
        </div>
      </div>
    
    
    
    <div id="content-wrapper">
      <div class="container">

        <div id="columns">
          <div class="columns-inner clearfix">

            <div id="content-column">
              <div class="content-inner">

                
                <section id="main-content" role="main">

                  
                  <div class="content-margin">
                    <div class="content-style">

                      
                      
                      
                                              <header class="clearfix">
                          <h1 id="page-title">
                            BioVenn - an R and Python package for the comparison and visualization of biological lists using area-proportional Venn diagrams                          </h1>
                        </header>
                      
                      
                      
                      <div id="content">
                        <div class="region region-content"><div id="block-system-main" class="block block-system no-title odd first last block-count-3 block-region-content block-main" >  
  
  <article id="node-679" class="node node-submit-paper article odd node-lang-en node-full ia-n clearfix" about="/paper/biovenn-r-and-python-package-comparison-and-visualization-biological-lists-using-area" typeof="sioc:Item foaf:Document" role="article">
  <div class="node-inner">



              <header class="node-header">

        
        
      </header>
        <span property="dc:title" content="BioVenn - an R and Python package for the comparison and visualization of biological lists using area-proportional Venn diagrams" class="rdf-meta element-hidden"></span><span property="sioc:num_replies" content="0" datatype="xsd:integer" class="rdf-meta element-hidden"></span>
    <div class="node-content">
        
      <h3>Tracking #: 679-1659</h3>          								<fieldset>
					<!-- legend>
						<span class="fieldset-legend">Paper Data</span>
					</legend -->

					


					
					<div style="background-color: #E3E5E3;border-radius: 5px;padding-top:4px;padding-bottom:4px;padding-left:6px;"><h5>Authors:</h5><br/><form action="biovenn-r-and-python-package-comparison-and-visualization-biological-lists-using-area.html" method="post" id="authors-table-form-anonymus" accept-charset="UTF-8"><div><div id="authors_Container"><table class="sticky-enabled">
 <thead><tr><th></th><th>Name</th><th>ORCID</th> </tr></thead>
<tbody>
 <tr class="odd"><td><div class="form-item form-type-checkbox form-item-authors-1365">
  <input type="checkbox" id="edit-authors-1365" name="authors[1365]" value="1365" class="form-checkbox" />
</div>
</td><td>Tim Hulsen</td><td><img alt="ORCID logo" src="../sites/default/files/orcid_24x24.png" style="vertical-align: middle; margin:0px;"/> <a href="https://orcid.org/0000-0002-0208-8443" target="_blank">https://orcid.org/0000-0002-0208-8443</a></td> </tr>
</tbody>
</table>
</div><input type="hidden" name="author_nd" value="679" />
<input type="hidden" name="author_vd" value="1659" />
<input type="hidden" name="form_build_id" value="form-qI8dAGybo2vx5krfkLF-YqIhVpWSywjc9mpg6obaZto" />
<input type="hidden" name="form_id" value="authors_table_form_anonymus" />
</div></form><hr/></div><br/>

					<div style="background-color: #E3E5E3;border-radius: 5px;padding-top:4px;padding-bottom:4px;padding-left:6px;"><section class="field field-name-field-editor field-type-user-reference field-label-above view-mode-full"><h2 class="field-label">Responsible editor:&nbsp;</h2><div class="field-items"><div class="field-item even">Gargi Datta</div></div></section></div><br/>
					                    <div style="background-color: #E3E5E3;border-radius:5px;padding-top:4px;padding-bottom:4px;padding-left:6px;">
					<section class="field field-name-field-submission-type field-type-list-text field-label-above view-mode-full"><h2 class="field-label">Submission Type:&nbsp;</h2><div class="field-items"><div class="field-item even">Resource Paper</div></div></section><section class="field field-name-field-abstract field-type-text-long field-label-above view-mode-full"><h2 class="field-label">Abstract:&nbsp;</h2><div class="field-items"><div class="field-item even">One of the most popular methods to visualize the overlap and differences between data sets is the Venn diagram. Venn diagrams are especially useful when they are &#039;area-proportional&#039; i.e. the sizes of the circles and the overlaps correspond to the sizes of the data sets. In 2007, the BioVenn web interface was launched, which is being used by many researchers. However, this web implementation requires users to copy and paste (or upload) lists of IDs into the web browser, which is not always convenient and makes it difficult for researchers to create Venn diagrams ‘in batch’, or to automatically update the diagram when the source data changes. This is only possible by using software such as R or Python. This paper describes the BioVenn R and Python packages, which are very easy-to-use packages that can generate accurate area-proportional Venn diagrams of two or three circles directly from lists of (biological) IDs. The only required input is two or three lists of IDs. Optional parameters include the main title, the subtitle, the printing of absolute numbers or percentages within the diagram, colors and fonts. The function can show the diagram on the screen, or it can write output to one of the supported file formats. The function also returns all thirteen lists. The BioVenn R package and Python package were created for biological IDs, but they can be used for other IDs as well. Finally, BioVenn can map Affymetrix and EntrezGene to Ensembl IDs. The BioVenn R package is available in the CRAN repository, and can be installed by running ‘install.packages(“BioVenn”)’. The BioVenn Python package is available in the PyPI repository, and can be installed by running ‘pip install BioVenn’. The BioVenn web interface remains available at https://www.biovenn.nl.</div></div></section><section class="field field-name-field-pdf field-type-file field-label-above view-mode-full"><h2 class="field-label">Manuscript:&nbsp;</h2><div class="field-items"><div class="field-item even"><span class="file"><img class="file-icon" alt="File" title="application/vnd.openxmlformats-officedocument.wordprocessingml.document" src="https://datasciencehub.net/modules/file/icons/x-office-document.png" /> <a href="../system/files/ds-paper-679.docx" type="application/vnd.openxmlformats-officedocument.wordprocessingml.document; length=9234129">ds-paper-679.docx</a></span></div></div></section><section class="field field-name-field-previous-version field-type-node-reference field-label-above view-mode-full"><h2 class="field-label">Previous Version:&nbsp;</h2><div class="field-items"><div class="field-item even"><a href="biovenn-–-r-and-python-package-comparison-and-visualization-biological-lists-using-area-0.html">BioVenn – an R and Python package for the comparison and visualization of biological lists using area-proportional Venn diagrams</a></div></div></section><section class="field field-name-field-tags field-type-taxonomy-term-reference field-label-above view-mode-full"><h2 class="field-label">Tags:&nbsp;</h2><ul class="field-items"><li class="field-item even">Reviewed</li></ul></section><section class="field field-name-field-data-repository-urls field-type-text-long field-label-above view-mode-full"><h2 class="field-label">Data repository URLs:&nbsp;</h2><div class="field-items"><div class="field-item even"><div class="field-items">
<div class="field-item even">
<p><a href="https://omabrowser.org/All/oma-groups.txt.gz">https://omabrowser.org/All/oma-groups.txt.gz</a></p>
<p><a href="https://www.biovenn.nl/r_python/">https://www.biovenn.nl/r_python/</a></p>
</div>
</div>
</div></div></section><section class="field field-name-field-date-of-submission field-type-datetime field-label-above view-mode-full"><h2 class="field-label">Date of Submission:&nbsp;</h2><div class="field-items"><div class="field-item even"><span class="date-display-single" property="dc:date" datatype="xsd:dateTime" content="2021-02-11T00:00:00-05:00">Thursday, February 11, 2021</span></div></div></section><section class="field field-name-field-date-of-decision field-type-datetime field-label-above view-mode-full"><h2 class="field-label">Date of Decision:&nbsp;</h2><div class="field-items"><div class="field-item even"><span class="date-display-single" property="dc:date" datatype="xsd:dateTime" content="2021-02-28T00:00:00-05:00">Sunday, February 28, 2021</span></div></div></section><br><br><b>Nanopublication URLs:</b> <br>                    </div>
                    <br/>

					
                    
                    <div style="background-color: #E3E5E3;border-radius: 5px;padding-top:4px;padding-bottom:4px;padding-left:6px;"><section class="field field-name-field-decision field-type-list-text field-label-above view-mode-full"><h2 class="field-label">Decision:&nbsp;</h2><div class="field-items"><div class="field-item even">Accept</div></div></section></div><br/>                    <div id="review">
                    		                    	</div>

					
										<div style="background-color: #E3E5E3;border-radius: 5px;padding-top:4px;padding-bottom:4px;padding-left:6px;"><b>Solicited Reviews:</b><br/><form action="biovenn-r-and-python-package-comparison-and-visualization-biological-lists-using-area.html" method="post" id="get-submitted-reviews" accept-charset="UTF-8"><div><fieldset class="collapsible collapsed form-wrapper" id="edit-fset1"><legend><span class="fieldset-legend">Review #1 <strong>submitted on 22/Feb/2021</strong></span></legend><div class="fieldset-wrapper with-legend"><div class="fieldset-description">By <strong>Emma Beauxis-Aussalet</strong> <img alt="ORCID logo" src="../sites/default/files/orcid_24x24.png" style="vertical-align: middle; margin:0px;"/> <a href="https://orcid.org/0000-0002-4657-892X" target="_blank">https://orcid.org/0000-0002-4657-892X</a><br/><br/>Review Details<br/><hr/></div><p>Reviewer has chosen <b>not</b> to be <strong><em>Anonymous</em></strong></p><b style="color:#f39222">Overall Impression:</b> Average</br><b style="color:#f39222">Suggested Decision: </b> Accept</br><b style="color:#f39222">Technical Quality of the paper:</b> Good<br/><b style="color:#f39222">Presentation:</b> Good<br/><b style="color:#f39222">Reviewer`s confidence:</b> Medium<br/><b style="color:#f39222">Significance:</b> Moderate significance<br/><b style="color:#f39222">Background:</b> Reasonable<br/><b style="color:#f39222">Novelty:</b> Limited novelty<br/><b style="color:#f39222">Data availability:</b> All used and produced data (if any) are FAIR and openly available in established data repositories<br/><b style="color:#f39222">Length of the manuscript:</b> The length of this manuscript is about right</p><p><b style="color:#f39222">Summary of paper in a few sentences (summary of changes and improvements for
second round reviews): </b><br/><p>The main changes is that the authors have clarified the mapping of "biological IDs", and introduced explanations of "the 13 sets" early in the paper's introduction.</p>
</p><p><b style="color:#f39222">Reasons to accept: </b><br/><p>The paper is more complete and easier to read.</p>
</p><p><b style="color:#f39222">Reasons to reject: </b><br/><p>IMPORTANT NOTE: the paper cannot be published as such, because some figures are missing (!) or are of poor quality in this version. However, in the previous version, the figure were fine. So I considered this a small mistake that will be easily corrected without a new round of revision.</p>
<p>Table 3 needs reformatting: it is too tedious to read as such.</p>
<p>The explanations of "the 13 sets" remains a bit vague (p.2). I'd suggest: "it displays the list of elements belonging to each of the subsets in the Venn diagram, i.e., the thirteen subsets resulting from the overlaps between two to three circles X, Y, Z (in other words, the elements belonging to all circles X, Y and Z; belonging to at least X, Y, Z, X and Y, X and Z, X and Z; belonging to only X, only Z, ... or only YZ).</p>
</p><p><b style="color:#f39222">Nanopublication comments: </b><br/></p><p><b style="color:#f39222">Further comments: </b><br/><p>Minor corrections:<br />
- Section 2.5: "himself" -&gt; "herself", or use plurals "unless users calculate ... themselves"<br />
- Section 5 (p.4): "it can write output to one of the ..." -&gt; "it can export the graph in one of the ..., e.g., as image or svg". And also "which is new" -&gt; "which is a new".</p>
</p></div></fieldset>
<fieldset class="collapsible collapsed form-wrapper" id="edit-fset2"><legend><span class="fieldset-legend">Review #2 <strong>submitted on 24/Feb/2021</strong></span></legend><div class="fieldset-wrapper with-legend"><div class="fieldset-description">By <strong>Michael Hinterberg</strong> <img alt="ORCID logo" src="../sites/default/files/orcid_24x24.png" style="vertical-align: middle; margin:0px;"/> <a href="https://orcid.org/0000-0003-0693-7075" target="_blank">https://orcid.org/0000-0003-0693-7075</a><br/><br/>Review Details<br/><hr/></div><p>Reviewer has chosen <b>not</b> to be <strong><em>Anonymous</em></strong></p><b style="color:#f39222">Overall Impression:</b> Good</br><b style="color:#f39222">Suggested Decision: </b> Accept</br><b style="color:#f39222">Technical Quality of the paper:</b> Good<br/><b style="color:#f39222">Presentation:</b> Good<br/><b style="color:#f39222">Reviewer`s confidence:</b> High<br/><b style="color:#f39222">Significance:</b> Moderate significance<br/><b style="color:#f39222">Background:</b> Reasonable<br/><b style="color:#f39222">Novelty:</b> Limited novelty<br/><b style="color:#f39222">Data availability:</b> All used and produced data (if any) are FAIR and openly available in established data repositories<br/><b style="color:#f39222">Length of the manuscript:</b> The length of this manuscript is about right</p><p><b style="color:#f39222">Summary of paper in a few sentences (summary of changes and improvements for
second round reviews): </b><br/><p>The manuscript describes the cross-platform capabilities of the BioVenn tool for visual comparison of gene lists. Through additional explanation (section 5.3) and clarification, the author has addressed concerns to improve the explanation of specific utility to biological analysis, as well as comparative strengths of the tool in drawing circles (as opposed to ellipses) and easy adjustment of visual parameters for production-quality plots.</p>
</p><p><b style="color:#f39222">Reasons to accept: </b><br/><p>The paper describes an example of a biological analysis for rapid hypothesis testing and visualization of gene set analysis.<br />
Thank you for the clarifying language and comments.</p>
</p><p><b style="color:#f39222">Reasons to reject: </b><br/><p>There are no major remaining concerns.</p>
</p><p><b style="color:#f39222">Nanopublication comments: </b><br/></p><p><b style="color:#f39222">Further comments: </b><br/></p></div></fieldset>
<input type="hidden" name="form_build_id" value="form-WHqAQVsb-bDfbfj2eMT0M5ZRnbr1UZ6VQy00ZqU_wJo" />
<input type="hidden" name="form_id" value="get_submitted_reviews" />
</div></form></div><br/>                    
					<div style="background-color: #E3E5E3;border-radius: 5px;padding-top:4px;padding-bottom:4px;padding-left:6px;"><fieldset class="collapsible collapsed group-response-to-reviews field-group-fieldset form-wrapper"><legend><span class="fieldset-legend">RESPONSE TO REVIEWERS</span></legend><div class="fieldset-wrapper with-legend"><div class="field field-name-field-response-to-reviews field-type-text-long field-label-hidden view-mode-full"><div class="field-items"><div class="field-item even"><p>Reviewer 1:</p>
<p>Summary of paper in a few sentences:</p>
<p>The manuscript describes the port of BioVenn to R and Python, which allows users to easily automate their operations with the algorithm.<br />
The work describes an update of a tool for proportional Venn diagrams, which has been widely used. The manuscript reviews existing tools for similar uses with fairness, showing strengths and weaknesses. I only have two small observations in this regard.<br />
--&gt; Thank you.<br />
• First, even though it is not properly a Venn diagram generator, the manuscript should mention UpSetR. I see no need to change the tables or figures, but a small description would improve the manuscript.<br />
--&gt; I have included UpSetR in the Conclusion section.<br />
• Second, nVenn has both an R interface (nVennR) and a web interface.<br />
--&gt; This has been corrected in table 3.</p>
<p>Regarding the technical merits of the manuscript:<br />
• One feature I consider unique and useful in BioVenn is the automatic mapping of Affymetrix and EntrezGene IDs to Ensembl IDs. However, I have not found this option while using the R interface, and it is not mentioned in the vignette. If this is not available, the author should delete the last sentence of the first paragraph of Results. If it is available, the author should add the procedure to the manuscript and the documentation.<br />
--&gt; I have included an example of the ID mapping in the manuscript (in the new section 5.3). It will also be added to the documentation of the next version of the software.<br />
• The manuscript should make it clear that exact proportionality for more than two sets is not achievable with circles. This is not a limitation of the program. In the example provided, if we delete the "1007_s_at" element in “list_z”, the “list_x intersect list_z” overlap should be zero. However, the overlap in the result is nonzero (the region is not labeled, which informs the user of this discrepancy). For the sake of completeness, this limitation should be pointed out in the manuscript.<br />
--&gt; Indeed, for three-circle diagrams it is sometimes not possible to have exact proportionality, and for diagrams with even more circles this is an even larger issue. I have added this information to both the Methods and the Results section.</p>
<p>Regarding presentation:<br />
• The text, figures, and tables of the work are accessible, pleasant to read, clearly structured, and free of major errors in grammar or style. I would point one small error at page 9: “This also makes sure that the user cannot mathematically impossible numbers…” should be “This also ensures that the user cannot input mathematically impossible numbers…”<br />
--&gt; This mistake has been corrected.</p>
<p>Reasons to accept:</p>
<p>The tool described is widely used and has some unique features that may be interesting to scientists.<br />
--&gt; Thank you.</p>
<p>Reasons to reject:</p>
<p>I do not find any issues important enough to reject the manuscript.<br />
--&gt; Thank you.</p>
<p>Reviewer 2:</p>
<p>Summary of paper in a few sentences:</p>
<p>Tim Hulsen provides an updated, straightforward R and Python package to support drawing of area-proportional Venn diagrams of up to 3 overlapping lists. Biological lists can be mapped from Entrez/Affy to Enembl before calculating overlaps. The functions are parameterized so layout elements such as labels, font, colour, etc. can be adjusted. Additionally, the functions return the overlapping list permutations.<br />
--&gt; Thank you.</p>
<p>Reasons to accept:</p>
<p>When working directly within R or Python, BioVenn appears to provide a quick and easy option for creating presentation and manuscript-worthy images. This is especially true within the specific use case of working with and between supported EntrezGene, Affymetrix, and Ensembl IDs. In those use cases, it is easier than using a web-based option, and the area-proportional output can be helpful for rapid visual insight.<br />
--&gt; Thank you.</p>
<p>Reasons to reject:</p>
<p>The novelty of this tool is the biggest drawback. It appears to be an API change to an existing tool more than a truly novel approach.<br />
--&gt; The R and Python packages were created by taking the source code of the original web application, and completely rewriting it in R and Python, taking into account the specific qualities and peculiarities of these programming languages. The mathematical calculations are the same in each version, making sure that they give the same output (except for some purely visual differences), but for the rest they are actually quite different. For example, to create SVG output, different modules/methods had to be used in the web version, the R package and the Python package. Furthermore, I would like to stress that some advancements have been made since the publication of 2008. Some errors have been corrected (caused by division-by-zero in some special cases), and the drag-and-drop functionality is new as well, making it much easier to put labels in the right spot (I included this information now in section 5).<br />
The "Bio" part is of limited use (or not properly explained), as a user with R, for example, could pre-process ID translations if desired, and a biological use-case is not strongly described in this paper.<br />
--&gt; The "Bio" part has been explained a better now in the paper (see the new section 5.3, including an extra figure and extra table). Indeed, ID mapping can be done separately beforehand in both R and Python, but it is just much more simple for the user to directly send the biological identifiers to BioVenn and let the program do the work of 1) the ID mapping, 2) the set/overlap calculation and 3) the venn diagram visualization.<br />
For general use, it is unclear if there are substantial differences between this and 'eulerr' for visual output in R. There are a few formatting parameters that appear to be supported in this tool (BioVenn), and of course it works natively in R and Python, but eulerr has a similar interface and allows for &gt;3 sets.<br />
--&gt; Eulerr indeed misses some formatting options that BioVenn has, and it does not have the drag-and-drop functionality to ensure that labels are places at the right spot. Allowing more than three sets is an interesting option (which I am actually exploring in new software), but it also brings additional problems since area-proportionality is often not possible for diagrams with more than three sets. Even for three there is not always a perfect solution, but with more circles this is a bigger problem.<br />
While the cross-platform support in R and Python may be helpful for some, a natural R improvement would be a tidy data/ggplot API that supported additional overlapping sets, as well as an easier layered graphical interface such that layout elements can easily be 'themed' instead of a lengthy and specific argument list.<br />
--&gt; Thank you for the suggestion. You stated correctly that 'cross-platform support in R and Python may be helpful for some'; therefore, any new functionality inplemented R should also be implemented in Python. I did not want to include any R-specific (or Python-specific) functionality. Considering the 'themes': users could easily store a list of parameters in a list/array/dictionary, and pass that on to the draw.venn function. Example:<br />
    black_theme 
    do.call(draw.venn,c(list(list_x,list_y,list_z),black_theme))<br />
Additionally, while the area-proportional plots seem intuitive and visually attractive, the paper could benefit for an insight-driven/workflow use case that showed the utility of area-proportionality; for example, real-world cases where equal-sized circles could be misleading, or where proportionality instead improved cognition.<br />
--&gt; Thank you for the suggestion. I believe that the figures 3 and 4 show clearly that the diagrams that show more overlap between human and mouse (opposed to human-xenopus and mouse-xenopus) are much more insightful. Also note the sentence "We can see that the packages that create area-proportional diagrams (a, c, d, g, h) give a better impression of what the data looks like: the human and mouse circles indeed have a larger overlap than with the Xenopus circle, and the Xenopus circle is larger than the other ones.". The diagrams that are not area-proportional do not show at a glance that human and mouse are relatively closely related, and that Xenopus has the largest genome.</p>
<p>In summary, this is an easy and potentially useful tool for specific use cases, with limited additional support compared to alternatives (although the alternatives are nicely shown). A scientific paper, as opposed to a Vignette/application note, would benefit more strongly from a specific use-case example or literature examples showing, or at least suggesting, the insight gained by presenting proportional Venn diagrams, which would be more compelling material for a paper than a description of all of the functional arguments.<br />
--&gt; Thank you. I have created section 5.3 to show a use case for the biological ID mapping. Sections 5.4. shows, in my opinion, clearly why area-proportional diagrams give better insight into the data, by showing the large human-mouse overlap and the large size of the Xenopus circle.</p>
<p>Reviewer 3:</p>
<p>Summary of paper in a few sentences:</p>
<p>The paper introduces R and python libraries for drawing Venn diagram, with areas proportional to the size of each set and subset.<br />
Compared to existing libraries, the new libraries include: automatic mapping of IDs into sets/subsets, choice of displaying set size as number or percentage, full adjustment of color and text aesthetics (incl. interactive label positioning, with SVG option). They also include a mapping of "Affymetrix and EntrezGene IDs to Ensembl IDs." which is not clearly specified.<br />
  --&gt; A better explanation of the ID mapping functionality has now been provided, including a figure and table created using the ID mapping.</p>
<p>Reasons to accept:</p>
<p>The novelty of the new libraries is clear, the new functionalities seem relevant for the bioinformatics community, and beyond. The review of existing libraries is clear and comprehensive. The paper is clear and well-written.<br />
--&gt; Thank you.</p>
<p>Reasons to reject:</p>
<p>The mapping of mapping of "Affymetrix and EntrezGene IDs to Ensembl IDs" remains unspecified, and a few other points need clarification (see below).<br />
It is important to briefly specify the mapping of "Affymetrix and EntrezGene IDs to Ensembl IDs" or else readers are not provided with all information needed to understand what the libraries does. If it is unclear what this special mapping does, then it is unclear how to reuse the libraries, especially for users who do not display bioinformatics data (and would not read citation [15] to fully understand the mapping). The potential reuse for use cases other than bioinformatics needs to be clarified (e.g., need to disable the "special" mapping? or just avoid ID labels in the typical bioinformatics format?).<br />
--&gt; An explanation of the ID mapping functionality has now been provided, in the new section 5.3. When users do not want to use this option, they can just leave out the 'map2ens' parameter, since its default is 'False'. This is now stated in the Methods section.</p>
<p>Further comments:</p>
<p>Direct links to the reference manuals should be added (e.g., for R <a href="https://cran.r-project.org/web/packages/BioVenn/BioVenn.pdf">https://cran.r-project.org/web/packages/BioVenn/BioVenn.pdf</a>).<br />
--&gt; CRAN itself states that the link should be <a href="https://CRAN.R-project.org/package=BioVenn">https://CRAN.R-project.org/package=BioVenn</a> (see the header 'Linking'). From there, the manual is just one click away. The same holds for the other packages.</p>
<p>Introduction:<br />
- It ends with " the following paragraphs" but describes only the next section. Having on overview of all sections would be clearer.<br />
  --&gt; I changed it to "the following two sections". Section 2 describes the R packages, section 3 the Python packages.<br />
- The "ID mapping" functionality should also be specified in the introduction, both the "basic mapping" into the "13 sets" and the mapping of "Affymetrix and EntrezGene IDs to Ensembl IDs" (e.g., described with a brief sentence).<br />
  --&gt; I have included a short description of both the biological ID mapping and the "13 sets" in the introduction.<br />
- Section 3.1: "‘drag-and-drop’ functionality of text and numbers" -&gt; "‘drag-and-drop’ functionality for repositioning labels"<br />
  --&gt; This has been changed to "'drag-and-drop' functionality for repositioning of titles and labels" (at multiple locations in the text).<br />
- Section 4: "Generate lists of the thirteen possible sets, and count them" -&gt; "and count the number of elements per set"<br />
  --&gt; Changed to "Generate lists of IDs for the thirteen possible sets, and count the number of IDs within each set"<br />
- Section 4: "Calculate the angles of the XYZ triangle" -&gt; specify what is the XYZ triangle (I understand that it's the centre of the 3 circles, but the reader is left to interpret what the "XYZ triangle" is).<br />
  --&gt; This is the triangle that is formed by connecting the three centers of the circles. This is now explained in the paper.<br />
- Section 5.3: "Xenopus" -&gt; "frog" for consistency with other labels of animals.<br />
  --&gt; "Xenopus" is a special type of frog (the African clawed frog), which is quite well known among researchers working with model organisms. Therefore the name "Xenopus" should be used here.<br />
- Table 2: Improve the layout<br />
  --&gt; The current layout of the table (now table 3) is not optimal due to limitations in MS Word. When it will be online, the layout will be much better.<br />
- Page 9: recap why areas in other libraries are "inaccurate". Also fix the syntax of "the user cannot mathematically impossible numbers" -&gt; "the user cannot [input] impossible numbers"<br />
  --&gt; I added a sentence explaining that for more than three sets, it is more difficult to create accurate area-proportional diagrams. The syntax has been corrected to "the user cannot input mathematically impossible numbers".</p>
</div></div></div></div></fieldset>
</div><br/>
				</fieldset>
				
    </div>

    <div id="admin-review">
			</div>

	<div id="decision">
			</div>




          <nav class="clearfix"><ul class="links inline"><li class="comment_forbidden first"><span><a href="../user/login%3Fdestination=comment%252Freply%252F679%23comment-form.html">Log in</a> to post comments</span></li><li class="camera_ready_files_field_paper last"></li></ul></nav>
    
    
  </div>
  </article>

  </div></div>                      </div>

                      
                    </div>
                  </div>

                </section>

                
              </div>
            </div>

                        
          </div>
        </div>

      </div>
    </div>

    
    
          <div id="footer-wrapper">
        <div class="container clearfix">
          <footer class="clearfix" role="contentinfo">
            <div class="region region-footer"><div class="region-inner clearfix"><div id="block-block-16" class="block block-block no-title odd first last block-count-4 block-region-footer block-16" ><div class="block-inner clearfix">  
  
  <div class="block-content content"><p style="text-align:center;"><a href="https://www.iospress.nl/disclaimer/" target="_blank">Disclaimer</a> | <a href="https://content.iospress.com/page/privacy" target="_blank">Privacy Policy</a> | Data Science is Published by <a href="http://www.iospress.com" target="_blank">IOS Press</a>, Copyright 2023</p>
</div>
  </div></div></div></div>            <p class="attribute-creator"></p>
          </footer>
        </div>
      </div>
    
  </div>
</div>
  </body>
</html>
