
<!DOCTYPE html>
<!--[if lt IE 7]><html class="lt-ie9 lt-ie8 lt-ie7" lang="en" dir="ltr"><![endif]-->
<!--[if IE 7]><html class="lt-ie9 lt-ie8" lang="en" dir="ltr"><![endif]-->
<!--[if IE 8]><html class="lt-ie9" lang="en" dir="ltr"><![endif]-->
<!--[if gt IE 8]><!--><html lang="en" dir="ltr" prefix="content: http://purl.org/rss/1.0/modules/content/ dc: http://purl.org/dc/terms/ foaf: http://xmlns.com/foaf/0.1/ og: http://ogp.me/ns# rdfs: http://www.w3.org/2000/01/rdf-schema# sioc: http://rdfs.org/sioc/ns# sioct: http://rdfs.org/sioc/types# skos: http://www.w3.org/2004/02/skos/core# xsd: http://www.w3.org/2001/XMLSchema#"><!--<![endif]-->
<head>
<meta charset="utf-8" />
<meta name="Generator" content="Drupal 7 (http://drupal.org)" />
<link rel="canonical" href="../paper/worcs-workflow-open-reproducible-code-science.html" />
<link rel="shortlink" href="../node/659.html" />
<link rel="shortcut icon" href="../sites/default/files/favicon-96x96.png" type="image/png" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta name="MobileOptimized" content="width" />
<meta name="HandheldFriendly" content="true" />
<meta name="apple-mobile-web-app-capable" content="yes" />
<title>WORCS: A Workflow for Open Reproducible Code in Science | Data Science</title>
<style type="text/css" media="all">
@import url("https://datasciencehub.net/modules/system/system.base.css?t4kojz");
@import url("https://datasciencehub.net/modules/system/system.menus.css?t4kojz");
@import url("https://datasciencehub.net/modules/system/system.messages.css?t4kojz");
@import url("https://datasciencehub.net/modules/system/system.theme.css?t4kojz");
</style>
<style type="text/css" media="all">
@import url("https://datasciencehub.net/misc/ui/jquery.ui.core.css?t4kojz");
@import url("https://datasciencehub.net/misc/ui/jquery.ui.theme.css?t4kojz");
</style>
<style type="text/css" media="all">
@import url("https://datasciencehub.net/modules/comment/comment.css?t4kojz");
@import url("https://datasciencehub.net/modules/field/theme/field.css?t4kojz");
@import url("../sites/all/modules/logintoboggan/logintoboggan.css%3Ft4kojz.css");
@import url("https://datasciencehub.net/modules/node/node.css?t4kojz");
@import url("https://datasciencehub.net/modules/search/search.css?t4kojz");
@import url("https://datasciencehub.net/modules/user/user.css?t4kojz");
@import url("../sites/all/modules/views/css/views.css%3Ft4kojz.css");
</style>
<style type="text/css" media="all">
@import url("../sites/all/modules/ctools/css/ctools.css%3Ft4kojz.css");
@import url("../sites/all/modules/ctools/css/modal.css%3Ft4kojz.css");
@import url("../sites/default/modules/modal_forms/css/modal_forms_popup.css%3Ft4kojz.css");
@import url("../sites/all/modules/responsive_menus/styles/responsive_menus_simple/css/responsive_menus_simple.css%3Ft4kojz.css");
@import url("../sites/all/modules/date/date_api/date.css%3Ft4kojz.css");
</style>
<style type="text/css" media="screen">
@import url("../sites/all/themes/adaptivetheme/at_core/css/at.settings.style.headings.css%3Ft4kojz.css");
@import url("../sites/all/themes/adaptivetheme/at_core/css/at.settings.style.image.css%3Ft4kojz.css");
@import url("../sites/all/themes/adaptivetheme/at_core/css/at.layout.css%3Ft4kojz.css");
</style>
<style type="text/css" media="all">
@import url("../sites/all/themes/corolla/css/html-elements.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/forms.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/tables.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/page.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/articles.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/comments.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/fields.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/blocks.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/navigation.css%3Ft4kojz.css");
@import url("../sites/all/themes/corolla/css/corolla.settings.style.css%3Ft4kojz.css");
@import url("../sites/default/files/color/corolla-80fb2671/colors.css%3Ft4kojz.css");
</style>
<style type="text/css" media="print">
@import url("../sites/all/themes/corolla/css/print.css%3Ft4kojz.css");
</style>
<link type="text/css" rel="stylesheet" href="../sites/default/files/adaptivetheme/corolla_files/corolla.responsive.layout.css%3Ft4kojz.css" media="only screen" />
<style type="text/css" media="screen">
@import url("../sites/default/files/adaptivetheme/corolla_files/corolla.fonts.css%3Ft4kojz.css");
</style>
<link type="text/css" rel="stylesheet" href="../sites/all/themes/corolla/css/responsive.custom.css%3Ft4kojz.css" media="only screen" />
<link type="text/css" rel="stylesheet" href="../sites/all/themes/corolla/css/responsive.smartphone.portrait.css%3Ft4kojz.css" media="only screen and (max-width:320px)" />
<link type="text/css" rel="stylesheet" href="../sites/all/themes/corolla/css/responsive.smartphone.landscape.css%3Ft4kojz.css" media="only screen and (min-width:321px) and (max-width:480px)" />
<link type="text/css" rel="stylesheet" href="../sites/all/themes/corolla/css/responsive.tablet.portrait.css%3Ft4kojz.css" media="only screen and (min-width:481px) and (max-width:768px)" />
<link type="text/css" rel="stylesheet" href="../sites/all/themes/corolla/css/responsive.tablet.landscape.css%3Ft4kojz.css" media="only screen and (min-width:769px) and (max-width:1024px)" />
<link type="text/css" rel="stylesheet" href="../sites/all/themes/corolla/css/responsive.desktop.css%3Ft4kojz.css" media="only screen and (min-width:1025px)" />

<!--[if lt IE 9]>
<style type="text/css" media="screen">
@import url("https://datasciencehub.net/sites/default/files/adaptivetheme/corolla_files/corolla.lt-ie9.layout.css?t4kojz");
</style>
<![endif]-->

<!--[if lte IE 9]>
<style type="text/css" media="screen">
@import url("https://datasciencehub.net/sites/all/themes/corolla/css/ie-lte-9.css?t4kojz");
</style>
<![endif]-->
<script type="text/javascript" src="https://datasciencehub.net/misc/jquery.js?v=1.4.4"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/jquery-extend-3.4.0.js?v=1.4.4"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/jquery-html-prefilter-3.5.0-backport.js?v=1.4.4"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/jquery.once.js?v=1.2"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/drupal.js?t4kojz"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/ui/jquery.ui.core.min.js?v=1.8.7"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/jquery.cookie.js?v=1.0"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/jquery.form.js?v=2.52"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/form-single-submit.js?v=7.98"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/form.js?v=7.98"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/ajax.js?v=7.98"></script>
<script type="text/javascript" src="../sites/default/modules/jquery_update/js/jquery_update.js%3Fv=0.0.1"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/progress.js?v=7.98"></script>
<script type="text/javascript" src="../sites/all/modules/ctools/js/modal.js%3Ft4kojz"></script>
<script type="text/javascript" src="../sites/default/modules/modal_forms/js/modal_forms_popup.js%3Ft4kojz"></script>
<script type="text/javascript" src="../sites/all/modules/responsive_menus/styles/responsive_menus_simple/js/responsive_menus_simple.js%3Ft4kojz"></script>
<script type="text/javascript" src="../sites/all/modules/google_analytics/googleanalytics.js%3Ft4kojz"></script>
<script type="text/javascript">
<!--//--><![CDATA[//><!--
(function(i,s,o,g,r,a,m){i["GoogleAnalyticsObject"]=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)})(window,document,"script","https://www.google-analytics.com/analytics.js","ga");ga("create", "UA-107867597-1", {"cookieDomain":"auto"});ga("set", "anonymizeIp", true);ga("send", "pageview");
//--><!]]>
</script>
<script type="text/javascript" src="../sites/all/modules/field_group/field_group.js%3Ft4kojz"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/tableheader.js?t4kojz"></script>
<script type="text/javascript" src="https://datasciencehub.net/misc/collapse.js?v=7.98"></script>
<script type="text/javascript">
<!--//--><![CDATA[//><!--
jQuery.extend(Drupal.settings, {"basePath":"\/","pathPrefix":"","setHasJsCookie":0,"ajaxPageState":{"theme":"corolla","theme_token":"id3HACQOW7VdDv-L4jt3DfKKzaxisUBOTcboNT8jLow","jquery_version":"default","js":{"misc\/jquery.js":1,"misc\/jquery-extend-3.4.0.js":1,"misc\/jquery-html-prefilter-3.5.0-backport.js":1,"misc\/jquery.once.js":1,"misc\/drupal.js":1,"misc\/ui\/jquery.ui.core.min.js":1,"misc\/jquery.cookie.js":1,"misc\/jquery.form.js":1,"misc\/form-single-submit.js":1,"misc\/form.js":1,"misc\/ajax.js":1,"sites\/default\/modules\/jquery_update\/js\/jquery_update.js":1,"misc\/progress.js":1,"sites\/all\/modules\/ctools\/js\/modal.js":1,"sites\/default\/modules\/modal_forms\/js\/modal_forms_popup.js":1,"sites\/all\/modules\/responsive_menus\/styles\/responsive_menus_simple\/js\/responsive_menus_simple.js":1,"sites\/all\/modules\/google_analytics\/googleanalytics.js":1,"0":1,"sites\/all\/modules\/field_group\/field_group.js":1,"misc\/tableheader.js":1,"misc\/collapse.js":1},"css":{"modules\/system\/system.base.css":1,"modules\/system\/system.menus.css":1,"modules\/system\/system.messages.css":1,"modules\/system\/system.theme.css":1,"misc\/ui\/jquery.ui.core.css":1,"misc\/ui\/jquery.ui.theme.css":1,"modules\/comment\/comment.css":1,"modules\/field\/theme\/field.css":1,"sites\/all\/modules\/logintoboggan\/logintoboggan.css":1,"modules\/node\/node.css":1,"modules\/search\/search.css":1,"modules\/user\/user.css":1,"sites\/all\/modules\/views\/css\/views.css":1,"sites\/all\/modules\/ctools\/css\/ctools.css":1,"sites\/all\/modules\/ctools\/css\/modal.css":1,"sites\/default\/modules\/modal_forms\/css\/modal_forms_popup.css":1,"sites\/all\/modules\/responsive_menus\/styles\/responsive_menus_simple\/css\/responsive_menus_simple.css":1,"sites\/all\/modules\/date\/date_api\/date.css":1,"sites\/all\/themes\/adaptivetheme\/at_core\/css\/at.settings.style.headings.css":1,"sites\/all\/themes\/adaptivetheme\/at_core\/css\/at.settings.style.image.css":1,"sites\/all\/themes\/adaptivetheme\/at_core\/css\/at.layout.css":1,"sites\/all\/themes\/corolla\/css\/html-elements.css":1,"sites\/all\/themes\/corolla\/css\/forms.css":1,"sites\/all\/themes\/corolla\/css\/tables.css":1,"sites\/all\/themes\/corolla\/css\/page.css":1,"sites\/all\/themes\/corolla\/css\/articles.css":1,"sites\/all\/themes\/corolla\/css\/comments.css":1,"sites\/all\/themes\/corolla\/css\/fields.css":1,"sites\/all\/themes\/corolla\/css\/blocks.css":1,"sites\/all\/themes\/corolla\/css\/navigation.css":1,"sites\/all\/themes\/corolla\/css\/fonts.css":1,"sites\/all\/themes\/corolla\/css\/corolla.settings.style.css":1,"sites\/all\/themes\/corolla\/color\/colors.css":1,"sites\/all\/themes\/corolla\/css\/print.css":1,"public:\/\/adaptivetheme\/corolla_files\/corolla.responsive.layout.css":1,"public:\/\/adaptivetheme\/corolla_files\/corolla.fonts.css":1,"sites\/all\/themes\/corolla\/css\/responsive.custom.css":1,"sites\/all\/themes\/corolla\/css\/responsive.smartphone.portrait.css":1,"sites\/all\/themes\/corolla\/css\/responsive.smartphone.landscape.css":1,"sites\/all\/themes\/corolla\/css\/responsive.tablet.portrait.css":1,"sites\/all\/themes\/corolla\/css\/responsive.tablet.landscape.css":1,"sites\/all\/themes\/corolla\/css\/responsive.desktop.css":1,"public:\/\/adaptivetheme\/corolla_files\/corolla.lt-ie9.layout.css":1,"sites\/all\/themes\/corolla\/css\/ie-lte-9.css":1}},"CToolsModal":{"loadingText":"Loading...","closeText":"Close Window","closeImage":"\u003Cimg typeof=\u0022foaf:Image\u0022 class=\u0022image-style-none\u0022 src=\u0022https:\/\/datasciencehub.net\/sites\/all\/modules\/ctools\/images\/icon-close-window.png\u0022 alt=\u0022Close window\u0022 title=\u0022Close window\u0022 \/\u003E","throbber":"\u003Cimg typeof=\u0022foaf:Image\u0022 class=\u0022image-style-none\u0022 src=\u0022https:\/\/datasciencehub.net\/sites\/all\/modules\/ctools\/images\/throbber.gif\u0022 alt=\u0022Loading\u0022 title=\u0022Loading...\u0022 \/\u003E"},"modal-popup-small":{"modalSize":{"type":"fixed","width":500,"height":150},"modalOptions":{"opacity":0.85,"background":"#000"},"animation":"fadeIn","modalTheme":"ModalFormsPopup","throbber":"\u003Cimg typeof=\u0022foaf:Image\u0022 class=\u0022image-style-none\u0022 src=\u0022https:\/\/datasciencehub.net\/sites\/default\/modules\/modal_forms\/images\/loading_animation.gif\u0022 alt=\u0022Loading...\u0022 title=\u0022Loading\u0022 \/\u003E","closeText":"Close"},"modal-popup-medium":{"modalSize":{"type":"fixed","width":550,"height":450},"modalOptions":{"opacity":0.85,"background":"#000"},"animation":"fadeIn","modalTheme":"ModalFormsPopup","throbber":"\u003Cimg typeof=\u0022foaf:Image\u0022 class=\u0022image-style-none\u0022 src=\u0022https:\/\/datasciencehub.net\/sites\/default\/modules\/modal_forms\/images\/loading_animation.gif\u0022 alt=\u0022Loading...\u0022 title=\u0022Loading\u0022 \/\u003E","closeText":"Close"},"modal-popup-large":{"modalSize":{"type":"scale","width":0.8,"height":0.8},"modalOptions":{"opacity":0.85,"background":"#000"},"animation":"fadeIn","modalTheme":"ModalFormsPopup","throbber":"\u003Cimg typeof=\u0022foaf:Image\u0022 class=\u0022image-style-none\u0022 src=\u0022https:\/\/datasciencehub.net\/sites\/default\/modules\/modal_forms\/images\/loading_animation.gif\u0022 alt=\u0022Loading...\u0022 title=\u0022Loading\u0022 \/\u003E","closeText":"Close"},"responsive_menus":[{"toggler_text":"\u2630 Menu","selectors":["#block-system-main-menu"],"media_size":"935","media_unit":"px","absolute":"1","remove_attributes":"1","responsive_menus_style":"responsive_menus_simple"}],"googleanalytics":{"trackOutbound":1,"trackMailto":1,"trackDownload":1,"trackDownloadExtensions":"7z|aac|arc|arj|asf|asx|avi|bin|csv|doc(x|m)?|dot(x|m)?|exe|flv|gif|gz|gzip|hqx|jar|jpe?g|js|mp(2|3|4|e?g)|mov(ie)?|msi|msp|pdf|phps|png|ppt(x|m)?|pot(x|m)?|pps(x|m)?|ppam|sld(x|m)?|thmx|qtm?|ra(m|r)?|sea|sit|tar|tgz|torrent|txt|wav|wma|wmv|wpd|xls(x|m|b)?|xlt(x|m)|xlam|xml|z|zip"},"urlIsAjaxTrusted":{"\/search\/node":true,"\/comment\/101":true},"field_group":{"fieldset":"full"},"ctools-sample-style":{"modalSize":{"type":"fixed","width":1098,"height":500,"addWidth":20,"addHeight":15},"modalOptions":{"opacity":0.5,"background-color":"#000"},"animation":"fadeIn","modalTheme":"CToolsSampleModal","throbber":"\u003Cimg typeof=\u0022foaf:Image\u0022 class=\u0022image-style-none\u0022 src=\u0022https:\/\/datasciencehub.net\/sites\/all\/modules\/ctools\/ctools_ajax_sample\/images\/ajax-loader.gif\u0022 alt=\u0022Loading...\u0022 title=\u0022Loading\u0022 \/\u003E"},"adaptivetheme":{"corolla":{"layout_settings":{"bigscreen":"three-col-grail","tablet_landscape":"three-col-grail","tablet_portrait":"one-col-vert","smalltouch_landscape":"one-col-vert","smalltouch_portrait":"one-col-stack"},"media_query_settings":{"bigscreen":"only screen and (min-width:1025px)","tablet_landscape":"only screen and (min-width:769px) and (max-width:1024px)","tablet_portrait":"only screen and (min-width:481px) and (max-width:768px)","smalltouch_landscape":"only screen and (min-width:321px) and (max-width:480px)","smalltouch_portrait":"only screen and (max-width:320px)"}}}});
//--><!]]>
</script>
<!--[if lt IE 9]>
<script src="https://datasciencehub.net/sites/all/themes/adaptivetheme/at_core/scripts/html5.js?t4kojz"></script>
<![endif]-->
</head>
<body class="html not-front not-logged-in no-sidebars page-node page-node- page-node-659 node-type-submit-paper site-name-hidden atr-7.x-3.x atv-7.x-3.1 lang-en site-name-data-science section-paper color-scheme-custom corolla bs-n bb-n mb-dd rc-6 rct-6">
  <div id="skip-link" class="nocontent">
    <a href="101.html#main-content" class="element-invisible element-focusable">Skip to main content</a>
  </div>
    <div id="page-wrapper">
  <div id="page" class="page ssc-n ssw-n ssa-l sss-n btc-n btw-b bta-l bts-n ntc-n ntw-b nta-l nts-n ctc-n ctw-b cta-l cts-n ptc-n ptw-b pta-l pts-n">

          <div id="menu-bar-wrapper">
        <div class="container clearfix">
          <div id="menu-bar" class="nav clearfix"><nav id="block-system-main-menu" class="block block-system block-menu menu-wrapper menu-bar-wrapper clearfix odd first block-count-1 block-region-menu-bar block-main-menu"  role="navigation">  
      <h2 class="element-invisible block-title">Main menu</h2>
  
  <ul class="menu clearfix"><li class="first leaf menu-depth-1 menu-item-757"><a href="../content/about-data-science.html">About</a></li><li class="leaf menu-depth-1 menu-item-758"><a href="../content/editors.html">Editorial Board</a></li><li class="leaf menu-depth-1 menu-item-1559"><a href="../content/submit-manuscript.html">Submit Manuscript</a></li><li class="leaf menu-depth-1 menu-item-1612"><a href="../content/calls.html">Calls</a></li><li class="leaf menu-depth-1 menu-item-759"><a href="../content/guidelines-authors.html">For Authors</a></li><li class="leaf menu-depth-1 menu-item-760"><a href="../content/guidelines-reviewers.html">For Reviewers</a></li><li class="leaf menu-depth-1 menu-item-582"><a href="../underreview.html" title="Under Review">Under Review</a></li><li class="leaf menu-depth-1 menu-item-583"><a href="../reviewed.html" title="">Reviewed</a></li><li class="last leaf menu-depth-1 menu-item-1611"><a href="../ReviewedAccepted.html" title="">Accepted</a></li></ul>
  </nav><div id="block-orcid-login-orcid-login" class="block block-orcid-login no-title menu-wrapper menu-bar-wrapper clearfix even last block-count-2 block-region-menu-bar block-orcid-login" >  
  
  <p style="background-color: #003b3a; float:right;"><a style="vertical-align: middle; color: white; padding: 10px; display: inline-block; text-decoration: none;" href="https://orcid.org/oauth/authorize?client_id=APP-223SGGDDO35LKOT2&amp;response_type=code&amp;scope=%2Fauthenticate&amp;redirect_uri=https%3A%2F%2Fdatasciencehub.net%2F&amp;state=eyJyZWRpcmVjdCI6ImNvbW1lbnRcLzEwMSJ9"><img alt="ORCID logo" src="../sites/default/files/orcid_24x24.png" style="vertical-align: middle;"/> Log In with ORCID</a></p>
  </div></div>        </div>
      </div>
    
    <div id="header-wrapper">
      <div class="container clearfix">
	        <header class="clearfix with-logo" role="banner">

                      <div id="branding" class="branding-elements clearfix">

                              <div id="logo">
                  <a href="../index.html"><img class="site-logo" typeof="foaf:Image" src="../sites/default/files/DS_LOGO_vertical.png" alt="Data Science" /></a>                </div>
              
                              <div class="element-invisible h-group" id="name-and-slogan">

                                      <h1 class="element-invisible" id="site-name"><a href="../index.html" title="Home page">Data Science</a></h1>
                  
                  
                </div>
              
            </div>
          
          
        </header>

      </div>
    </div>

          <div id="top-panels-wrapper">
        <div class="container clearfix">
          <!-- Three column 3x33 Gpanel -->
          <div class="at-panel gpanel panel-display three-3x33 clearfix">
                                                          </div>
        </div>
      </div>
    
    
    
    <div id="content-wrapper">
      <div class="container">

        <div id="columns">
          <div class="columns-inner clearfix">

            <div id="content-column">
              <div class="content-inner">

                
                <section id="main-content" role="main">

                  
                  <div class="content-margin">
                    <div class="content-style">

                      
                      
                      
                                              <header class="clearfix">
                          <h1 id="page-title">
                            WORCS: A Workflow for Open Reproducible Code in Science                          </h1>
                        </header>
                      
                      
                      
                      <div id="content">
                        <div class="region region-content"><div id="block-system-main" class="block block-system no-title odd first last block-count-3 block-region-content block-main" >  
  
  <article id="node-659" class="node node-submit-paper article odd node-lang-en node-full ia-n clearfix" about="/paper/worcs-workflow-open-reproducible-code-science" typeof="sioc:Item foaf:Document" role="article">
  <div class="node-inner">



              <header class="node-header">

        
        
      </header>
        <span property="dc:title" content="WORCS: A Workflow for Open Reproducible Code in Science" class="rdf-meta element-hidden"></span><span property="sioc:num_replies" content="2" datatype="xsd:integer" class="rdf-meta element-hidden"></span>
    <div class="node-content">
        
      <h3>Tracking #: 659-1639</h3>          								<fieldset>
					<!-- legend>
						<span class="fieldset-legend">Paper Data</span>
					</legend -->

					


					
					<div style="background-color: #E3E5E3;border-radius: 5px;padding-top:4px;padding-bottom:4px;padding-left:6px;"><h5>Authors:</h5><br/><form action="101.html" method="post" id="authors-table-form-anonymus" accept-charset="UTF-8"><div><div id="authors_Container"><table class="sticky-enabled">
 <thead><tr><th></th><th>Name</th><th>ORCID</th> </tr></thead>
<tbody>
 <tr class="odd"><td><div class="form-item form-type-checkbox form-item-authors-1706">
  <input type="checkbox" id="edit-authors-1706" name="authors[1706]" value="1706" class="form-checkbox" />
</div>
</td><td>Caspar Van Lissa</td><td><img alt="ORCID logo" src="../sites/default/files/orcid_24x24.png" style="vertical-align: middle; margin:0px;"/> <a href="https://orcid.org/0000-0002-0808-5024" target="_blank">https://orcid.org/0000-0002-0808-5024</a></td> </tr>
 <tr class="even"><td><div class="form-item form-type-checkbox form-item-authors-1707">
  <input type="checkbox" id="edit-authors-1707" name="authors[1707]" value="1707" class="form-checkbox" />
</div>
</td><td>Andreas M. Brandmaier</td><td><img alt="ORCID logo" src="../sites/default/files/orcid_24x24.png" style="vertical-align: middle; margin:0px;"/> <a href="https://orcid.org/0000-0001-8765-6982" target="_blank">https://orcid.org/0000-0001-8765-6982</a></td> </tr>
 <tr class="odd"><td><div class="form-item form-type-checkbox form-item-authors-1708">
  <input type="checkbox" id="edit-authors-1708" name="authors[1708]" value="1708" class="form-checkbox" />
</div>
</td><td>Loek Brinkman</td><td><img alt="ORCID logo" src="../sites/default/files/orcid_24x24.png" style="vertical-align: middle; margin:0px;"/> <a href="https://orcid.org/0000-0003-3997-1173" target="_blank">https://orcid.org/0000-0003-3997-1173</a></td> </tr>
 <tr class="even"><td><div class="form-item form-type-checkbox form-item-authors-1512">
  <input type="checkbox" id="edit-authors-1512" name="authors[1512]" value="1512" class="form-checkbox" />
</div>
</td><td>Anna-Lena Lamprecht</td><td><img alt="ORCID logo" src="../sites/default/files/orcid_24x24.png" style="vertical-align: middle; margin:0px;"/> <a href="https://orcid.org/0000-0003-1953-5606" target="_blank">https://orcid.org/0000-0003-1953-5606</a></td> </tr>
 <tr class="odd"><td><div class="form-item form-type-checkbox form-item-authors-1709">
  <input type="checkbox" id="edit-authors-1709" name="authors[1709]" value="1709" class="form-checkbox" />
</div>
</td><td>Aaron Peikert</td><td><img alt="ORCID logo" src="../sites/default/files/orcid_24x24.png" style="vertical-align: middle; margin:0px;"/> <a href="https://orcid.org/0000-0001-7813-818X" target="_blank">https://orcid.org/0000-0001-7813-818X</a></td> </tr>
 <tr class="even"><td><div class="form-item form-type-checkbox form-item-authors-1710">
  <input type="checkbox" id="edit-authors-1710" name="authors[1710]" value="1710" class="form-checkbox" />
</div>
</td><td>Marijn E. Struiksma</td><td><img alt="ORCID logo" src="../sites/default/files/orcid_24x24.png" style="vertical-align: middle; margin:0px;"/> <a href="https://orcid.org/0000-0002-1166-1424" target="_blank">https://orcid.org/0000-0002-1166-1424</a></td> </tr>
 <tr class="odd"><td><div class="form-item form-type-checkbox form-item-authors-1711">
  <input type="checkbox" id="edit-authors-1711" name="authors[1711]" value="1711" class="form-checkbox" />
</div>
</td><td>Barbara Vreede</td><td><img alt="ORCID logo" src="../sites/default/files/orcid_24x24.png" style="vertical-align: middle; margin:0px;"/> <a href="https://orcid.org/0000-0002-5023-4601" target="_blank">https://orcid.org/0000-0002-5023-4601</a></td> </tr>
</tbody>
</table>
</div><input type="hidden" name="author_nd" value="659" />
<input type="hidden" name="author_vd" value="1639" />
<input type="hidden" name="form_build_id" value="form-sULSIl4rix5KiI3Yzg9TjQ2FcKkEGB-RWSGFJ29KOHQ" />
<input type="hidden" name="form_id" value="authors_table_form_anonymus" />
</div></form><hr/></div><br/>

					<div style="background-color: #E3E5E3;border-radius: 5px;padding-top:4px;padding-bottom:4px;padding-left:6px;"><section class="field field-name-field-editor field-type-user-reference field-label-above view-mode-full"><h2 class="field-label">Responsible editor:&nbsp;</h2><div class="field-items"><div class="field-item even">Tobias Kuhn</div></div></section></div><br/>
					                    <div style="background-color: #E3E5E3;border-radius:5px;padding-top:4px;padding-bottom:4px;padding-left:6px;">
					<section class="field field-name-field-submission-type field-type-list-text field-label-above view-mode-full"><h2 class="field-label">Submission Type:&nbsp;</h2><div class="field-items"><div class="field-item even">Resource Paper</div></div></section><section class="field field-name-field-abstract field-type-text-long field-label-above view-mode-full"><h2 class="field-label">Abstract:&nbsp;</h2><div class="field-items"><div class="field-item even">Adopting open science principles can be challenging and time-intensive, because doing so requires substantial conceptual education and training in the use of new tools. This paper introduces the Workflow for Open Reproducible Code in Science (WORCS): A step-by-step procedure that researchers can follow to make a research project open and reproducible, accompanied by an open source software implementation that automates these steps. The purpose of the workflow is to lower the threshold for adoption of open science principles. It is based on best practices, and can be used either in parallel to, or in absence of, top-down requirements by journals, institutions, and funding bodies. This workflow is primarily targeted towards scholars working in R on a private computer, conducting research that involves academic prose, analysis code, and (optionally) tabular data - but it is flexible enough to accommodate many other scenarios. To facilitate widespread adoption, the WORCS principles are implemented in the R package worcs, which offers an RStudio project template that is, effectively, a &quot;one-click solution&quot; for R users, and utility functions for specific workflow steps. This paper addresses the design of the workflow, discusses how it meets different standards for open science, and provides examples of the implementation of worcs in R. All source code for the R package, and replication materials for the manuscript, are available at https://github.com/cjvanlissa/worcs.</div></div></section><section class="field field-name-field-pdf field-type-file field-label-above view-mode-full"><h2 class="field-label">Manuscript:&nbsp;</h2><div class="field-items"><div class="field-item even"><span class="file"><img class="file-icon" alt="Package icon" title="application/zip" src="https://datasciencehub.net/modules/file/icons/package-x-generic.png" /> <a href="../system/files/ds-paper-659.zip" type="application/zip; length=618973">ds-paper-659.zip</a></span></div></div></section><div class="view view-revised-verision view-id-revised_verision view-display-id-entity_view_1 view-dom-id-0784e907793ef3bf7b7df773e99b1f3b">
      
      <div class="view-header">
      <b>Revised Version:</b>    </div>
  
  
  
      <div class="view-content">
        <div class="views-row views-row-1 views-row-odd views-row-first views-row-last">
      
  <div class="views-field views-field-title">        <span class="field-content"><a href="../paper/worcs-workflow-open-reproducible-code-science-0.html">WORCS: A Workflow for Open Reproducible Code in Science</a></span>  </div>  </div>
    </div>
  
  
  
  
  
  
</div> <section class="field field-name-field-tags field-type-taxonomy-term-reference field-label-above view-mode-full"><h2 class="field-label">Tags:&nbsp;</h2><ul class="field-items"><li class="field-item even">Reviewed</li></ul></section><section class="field field-name-field-data-repository-urls field-type-text-long field-label-above view-mode-full"><h2 class="field-label">Data repository URLs:&nbsp;</h2><div class="field-items"><div class="field-item even"><p>none</p>
</div></div></section><section class="field field-name-field-date-of-submission field-type-datetime field-label-above view-mode-full"><h2 class="field-label">Date of Submission:&nbsp;</h2><div class="field-items"><div class="field-item even"><span class="date-display-single" property="dc:date" datatype="xsd:dateTime" content="2020-09-25T00:00:00-04:00">Friday, September 25, 2020</span></div></div></section><section class="field field-name-field-date-of-decision field-type-datetime field-label-above view-mode-full"><h2 class="field-label">Date of Decision:&nbsp;</h2><div class="field-items"><div class="field-item even"><span class="date-display-single" property="dc:date" datatype="xsd:dateTime" content="2020-10-27T00:00:00-04:00">Tuesday, October 27, 2020</span></div></div></section><br><br><b>Nanopublication URLs:</b> <br>                    </div>
                    <br/>

					
                    
                    <div style="background-color: #E3E5E3;border-radius: 5px;padding-top:4px;padding-bottom:4px;padding-left:6px;"><section class="field field-name-field-decision field-type-list-text field-label-above view-mode-full"><h2 class="field-label">Decision:&nbsp;</h2><div class="field-items"><div class="field-item even">Undecided</div></div></section></div><br/>                    <div id="review">
                    		                    	</div>

					
										<div style="background-color: #E3E5E3;border-radius: 5px;padding-top:4px;padding-bottom:4px;padding-left:6px;"><b>Solicited Reviews:</b><br/><form action="101.html" method="post" id="get-submitted-reviews" accept-charset="UTF-8"><div><fieldset class="collapsible collapsed form-wrapper" id="edit-fset1"><legend><span class="fieldset-legend">Review #1 <strong>submitted on 19/Oct/2020</strong></span></legend><div class="fieldset-wrapper with-legend"><div class="fieldset-description">By <strong>Daniel Garijo</strong> <br/><br/>Review Details<br/><hr/></div><p>Reviewer has chosen <b>not</b> to be <strong><em>Anonymous</em></strong></p><b style="color:#f39222">Overall Impression:</b> Weak</br><b style="color:#f39222">Suggested Decision: </b> Reject</br><b style="color:#f39222">Technical Quality of the paper:</b> Good<br/><b style="color:#f39222">Presentation:</b> Good<br/><b style="color:#f39222">Reviewer`s confidence:</b> High<br/><b style="color:#f39222">Significance:</b> High significance<br/><b style="color:#f39222">Background:</b> Reasonable<br/><b style="color:#f39222">Novelty:</b> Limited novelty<br/><b style="color:#f39222">Data availability:</b> All used and produced data (if any) are FAIR and openly available in established data repositories<br/><b style="color:#f39222">Length of the manuscript:</b> The length of this manuscript is about right</p><p><b style="color:#f39222">Summary of paper in a few sentences: </b><br/><p>This paper describes worcs, a package and workflow to help make R code reproducible according to the TOP guidelines.</p>
<p>The paper describes a tutorial of the tool, and how it implements the TOP guidelines.</p>
</p><p><b style="color:#f39222">Reasons to accept: </b><br/><p>- The paper is relevant to the journal and on a timely topic.<br />
- The capabilities of the tool for synthetic data generation, connect to OSF, preprint publishing and dependency gathering show a lot of promise.<br />
- All the software (except a 404 page in the help, which is minor) is available online; and the project seems maintained and well documented.</p>
</p><p><b style="color:#f39222">Reasons to reject: </b><br/><p>- There is no user evaluation, and therefore all the claims about worcs are unsupported by evidence. If the authors had included a compelling evaluation I would be in favor of accepting the paper.</p>
</p><p><b style="color:#f39222">Nanopublication comments: </b><br/></p><p><b style="color:#f39222">Further comments: </b><br/><p>This paper describes worcs, a package and workflow to help make R code reproducible according to the TOP guidelines.</p>
<p>The paper is well written, easy to follow and very relevant and timely for the journal. I think the capabilities of the worcs package for synthetic data generation, connect to osf, help with the preprints and capture dependencies are very valuable to the community. However, the main limitation of the paper is that the workflow and proposed package are not evaluated with users, and therefore the claims of the authors with respect of its capabilities are not supported by evidence. I would like to encourage the authors to gather user/community feedback and demonstrate with numbers that worcs can help scientists as they claim. Then, the paper would become a strong contribution for the journal.</p>
<p>Below I describe other points that could be improved in the current publication:</p>
<p>- Unsupported claims (beyond the ones derived from an absence of evaluation): The paper claims in many parts that worcs helps make an entire research project Open and Reproducible. However, at the same time there is also a claim for not supporting replication. Since replication is generally understood as a necessary steps towards reproducibility, I find this kind of contradictory. Maybe the claims can be modified stating that worcs helps documentation and understanding, hence supporting FAIR.</p>
<p>- I am confused by the citation "Van Lissa, Caspar J., Aaron Peikert, and Andreas M. Brandmaier. 2020. “Worcs: Workflow for Open Reproducible Code in Science.”". Is this referring to another paper or the actual package? If it's another paper then what is the contribution of this publication?</p>
<p>- If the paper is generated as DDG, where are the functions that show it's capabilities? It is also not clear to me that some of the functionalities of Latex would be usable in RMarkdown, such as all the equations.</p>
<p>- It is not clear how worcs addresses depedency management. The section lists a series of tools that can be used, but the section is not very specific to worcs.</p>
<p>- The workflow of releasing data products on GitHub is not ideal. First, because datasets of 100 MBs usually require special handling. Second, it doesn't follow very well the FAIR principles for findability and attribution of data. Instead, repositories like Zenodo or FigShare could be user to address data storage, with proper metadata and its corresponding citation. The authors seem to not describe very well what would happen in data science experiments where the data size is significant. </p>
<p>- In the 3 phases listed, it is not clear what worcs does and does not for users.</p>
<p>- In some cases it looks like the authors claim that everyone should use R for open science. While R has a strong community behind, I believe there are others with very strong support (e.g. Python). I am not sure if suggesting a shift to R is the right approach. Note that this is the impression I got from reading the paper, maybe it was unintended by the authors.</p>
<p>- Vignete on citation leads to a 404.</p>
<p>- Synthetic data generation is fantastic, but it is not clear to me how does worcs actually keep the data consistent e.g. for training models. Does it follow a similar distribution to the source data?</p>
<p>- Docker is deemed as complicated for novice users, but the setup vignete looks quite complex to me as well. Wouldn't a Docker container with worcs installed be easier to use? Maybe these aspects would be better highlighted in a user evaluation.</p>
</p></div></fieldset>
<fieldset class="collapsible collapsed form-wrapper" id="edit-fset2"><legend><span class="fieldset-legend">Review #2 <strong>submitted on 19/Oct/2020</strong></span></legend><div class="fieldset-wrapper with-legend"><div class="fieldset-description">By <strong>Remzi Celebi</strong> <img alt="ORCID logo" src="../sites/default/files/orcid_24x24.png" style="vertical-align: middle; margin:0px;"/> <a href="https://orcid.org/0000-0001-7769-4272" target="_blank">https://orcid.org/0000-0001-7769-4272</a><br/><br/>Review Details<br/><hr/></div><p>Reviewer has chosen <b>not</b> to be <strong><em>Anonymous</em></strong></p><b style="color:#f39222">Overall Impression:</b> Good</br><b style="color:#f39222">Suggested Decision: </b> Accept</br><b style="color:#f39222">Technical Quality of the paper:</b> Good<br/><b style="color:#f39222">Presentation:</b> Good<br/><b style="color:#f39222">Reviewer`s confidence:</b> Medium<br/><b style="color:#f39222">Significance:</b> High significance<br/><b style="color:#f39222">Background:</b> Reasonable<br/><b style="color:#f39222">Novelty:</b> Limited novelty<br/><b style="color:#f39222">Data availability:</b> With exceptions that are admissible according to the data availability guidelines, all used and produced data are FAIR and openly available in established data repositories<br/><b style="color:#f39222">Length of the manuscript:</b> The length of this manuscript is about right</p><p><b style="color:#f39222">Summary of paper in a few sentences: </b><br/><p>The paper introduces the Workflow for Open Reproducible Code in Science (WORCS) , a workflow that complies with most of best practices for open sciences projects and demonstrates the use of the R worcs package for adapting WORCS workflow. The package provides a template for R users to create projects that follow open science principles.</p>
</p><p><b style="color:#f39222">Reasons to accept: </b><br/><p>It provides a good discussion of TOP-guidelines (a set of open science principles) and how proposed workflow/package can be used to facilitate the adaptation of these guidelines. </p>
</p><p><b style="color:#f39222">Reasons to reject: </b><br/><p>I would suggest the authors include a flowchart for users that shows the decision making process for which tools/operations should be used in WORCS. An example project would also help users better understand the package.</p>
<p>The claim that WORCS is amenable to the FAIR Guiding Principles should be justified. What are the solutions provided by WORCS that make a project interoperable and reusable? It would be improved with a more thorough discussion on how WORCS meets each FAIR principle. </p>
<p>Is WORCS providing any service for de-identification? </p>
</p><p><b style="color:#f39222">Nanopublication comments: </b><br/></p><p><b style="color:#f39222">Further comments: </b><br/></p></div></fieldset>
<fieldset class="collapsible collapsed form-wrapper" id="edit-fset3"><legend><span class="fieldset-legend">Review #3 <strong>submitted on 26/Oct/2020</strong></span></legend><div class="fieldset-wrapper with-legend"><div class="fieldset-description">Review Details<br/><hr/></div><p>Reviewer has chosen to be <strong><em>Anonymous</em></strong></p><b style="color:#f39222">Overall Impression:</b> Weak</br><b style="color:#f39222">Suggested Decision: </b> Reject</br><b style="color:#f39222">Technical Quality of the paper:</b> Average<br/><b style="color:#f39222">Presentation:</b> Good<br/><b style="color:#f39222">Reviewer`s confidence:</b> High<br/><b style="color:#f39222">Significance:</b> High significance<br/><b style="color:#f39222">Background:</b> Reasonable<br/><b style="color:#f39222">Novelty:</b> Limited novelty<br/><b style="color:#f39222">Data availability:</b> All used and produced data (if any) are FAIR and openly available in established data repositories<br/><b style="color:#f39222">Length of the manuscript:</b> The length of this manuscript is about right</p><p><b style="color:#f39222">Summary of paper in a few sentences: </b><br/><p>The paper presents WORCS, a library that is implemented within R-studio with the purpose of promoting the creation of reproducible research. In particular, they assist the researchers in the tasks of generating the scholarly article, thereby managing a number of tasks that researchers have to deal with from the reformulation of their hypothesis to version control, to dependency management, and finally generating the scholarly article.</p>
</p><p><b style="color:#f39222">Reasons to accept: </b><br/><p>On the positive side, the authors built their tool, with Open Science principles in mind. In doing so, they have pined the actions that need to be performed as part of the research process. For instance, steps such as the pre-registration of the objective of the study before starting the investigation is ignored by existing solutions.</p>
</p><p><b style="color:#f39222">Reasons to reject: </b><br/><p>The objective of the author is ambiguous. That said, I do not think that WORCS provide sufficient features that make it stands out from state of the art tools, for the following reasons.</p>
<p>C1) The process as described by the author for conducting research is assumed to be linear. However, researchers often go back and forth on steps that they have already completed, and sometimes have to undo them completely and start from scratch. This does not seem to be tackled by the tool.</p>
<p>C2) The gist of the tool seems to allow researchers to document their steps and generate a scholarly document that is ready for submission. In a sense, there is a similarity with existing notebooks such as Jupyter. So I was wondering why don’t the authors start from an existing popular tool such as Jupyter as the starting point (given the similarity in features that they would like to add), and extend Jupyter for instance with the new features such as dependency documentation and version control ?</p>
<p>C3) Regarding Version control and dependency documentation, the authors uses in their system existing solution, namely git and env, respectively. While I did not criticize reusing existing solution, on the contrary, I don't believe that simply using such solution add a substantial value. For example, git capture changes at the level of the line of code/text, whereas a researcher needs an abstraction that informs him/her on the coarse-grained elements that are significant from the point of view of development and research steps so as to have a clear idea of where s/he is within the process and help him/her make the decision of the next steps to undertake.</p>
<p>To sum up, I think that the initial idea is interesting. However, I do not think that WORCS will be adopted by researchers or promote the state of the art of reproducibility, even if we focus on researchers who conduct their development and analysis using R.</p>
</p><p><b style="color:#f39222">Nanopublication comments: </b><br/></p><p><b style="color:#f39222">Further comments: </b><br/></p></div></fieldset>
<fieldset class="collapsible collapsed form-wrapper" id="edit-fset4"><legend><span class="fieldset-legend">Review #4 <strong>submitted on 26/Oct/2020</strong></span></legend><div class="fieldset-wrapper with-legend"><div class="fieldset-description">By <strong>Katherine Wolstencroft</strong> <img alt="ORCID logo" src="../sites/default/files/orcid_24x24.png" style="vertical-align: middle; margin:0px;"/> <a href="https://orcid.org/0000-0002-1279-5133" target="_blank">https://orcid.org/0000-0002-1279-5133</a><br/><br/>Review Details<br/><hr/></div><p>Reviewer has chosen <b>not</b> to be <strong><em>Anonymous</em></strong></p><b style="color:#f39222">Overall Impression:</b> Good</br><b style="color:#f39222">Suggested Decision: </b> Accept</br><b style="color:#f39222">Technical Quality of the paper:</b> Excellent<br/><b style="color:#f39222">Presentation:</b> Good<br/><b style="color:#f39222">Reviewer`s confidence:</b> Medium<br/><b style="color:#f39222">Significance:</b> Moderate significance<br/><b style="color:#f39222">Background:</b> Reasonable<br/><b style="color:#f39222">Novelty:</b> Limited novelty<br/><b style="color:#f39222">Data availability:</b> All used and produced data (if any) are FAIR and openly available in established data repositories<br/><b style="color:#f39222">Length of the manuscript:</b> The length of this manuscript is about right</p><p><b style="color:#f39222">Summary of paper in a few sentences: </b><br/><p>This paper describes a Workflow for Open Reproducible Code in Science (WORCS). It promotes reproducible and FAIR R code and complies with the TOP-guidelines. The workflow is presented with an R library and a Github repository containing worcs templates and all supporting materials for this manuscript.</p>
</p><p><b style="color:#f39222">Reasons to accept: </b><br/><p>The paper is well written and clearly describes WORCS, which promotes reproducible and open code for scientific investigations. It complies with current best practices for open and reproducible code and it also follows the FAIR principles. Such initiatives help promote better open and reproducible science. The approach taken here is pragmatic and lightweight, which is necessary for large scale uptake. The authors also focus on ease of use and provide a "one-click solution".</p>
<p>WORCS is for scientists working with R with RMarkdown. It could therefore be argued that the utility is limited. However, R is widely used across science and the approach taken here could serve as an example for other software.</p>
</p><p><b style="color:#f39222">Reasons to reject: </b><br/><p>The abstract of the paper states that  the manuscript "provides examples of the implementation of worcs in R", so I was expecting science projects that had been described using worcs. If these examples exist, they should be referenced and linked to from the manuscript, as they would provide a better demonstration of the practical utility of the workflow and software than tutorial style examples.</p>
<p>There is a comparison in the paper to another R-based reproducible science solution (Peikert and Brandmaier (2019)). The authors discuss some of the differences between these approaches, but the comparison is limited. It is not clear to me how different these approaches actually are, apart from ease of use. If scientists are already familiar with R, is there a significant learning curve with either solution?</p>
</p><p><b style="color:#f39222">Nanopublication comments: </b><br/></p><p><b style="color:#f39222">Further comments: </b><br/></p></div></fieldset>
<input type="hidden" name="form_build_id" value="form--iMnEIeIROWQ_WhDILqgxeEg_0bjt1pmvBmoh8fqVIo" />
<input type="hidden" name="form_id" value="get_submitted_reviews" />
</div></form></div><br/>                    
					<div style="background-color: #E3E5E3;border-radius: 5px;padding-top:4px;padding-bottom:4px;padding-left:6px;"></div><br/>
				</fieldset>
				
    </div>

    <div id="admin-review">
			</div>

	<div id="decision">
			</div>




          <nav class="clearfix"><ul class="links inline"><li class="comment_forbidden first"><span><a href="../user/login%3Fdestination=comment%252Freply%252F659%23comment-form.html">Log in</a> to post comments</span></li><li class="camera_ready_files_field_paper last"></li></ul></nav>
    
    <section id="comments" class="comment-wrapper">

          <h2 class="title">
      2 Comments    </h2>
      
  <a id="comment-100"></a>
<article class="comment odd first clearfix" about="/comment/100#comment-100" typeof="sioc:Post sioct:Comment">

  
  
      <header class="comment-header">
 
          <h3 property="dc:title" datatype="" class="comment-title"><a href="100.html#comment-100" rel="bookmark">Meta-Review by Editor</a></h3>
    
        
    <p class="comment-submitted"><span property="dc:date dc:created" content="2020-10-27T02:04:24-04:00" datatype="xsd:dateTime" rel="sioc:has_creator">Submitted by <span class="username" xml:lang="" about="/users/tobias-kuhn" typeof="sioc:UserAccount" property="foaf:name" datatype="">Tobias Kuhn</span> on <time datetime="2020-10-27T02:04:24Z"><span class="date-time">Tue, 10/27/2020 - 02:04</span></time></span><p>
  </header>
    
  <div class="comment-content">
    <span rel="sioc:reply_of" resource="/paper/worcs-workflow-open-reproducible-code-science" class="rdf-meta element-hidden"></span><div class="field field-name-comment-body field-type-text-long field-label-hidden view-mode-full"><div class="field-items"><div class="field-item even" property="content:encoded"><p>The reviewers agree that the paper has merit, but some of them also point to major shortcomings, in particular the lack of discussion of user adoption and of handling non-linear research processes. Note that according to our guidelines, resource papers do not need to have a full-blown user evaluation, but we expect "sound evidence of its (potential for) reuse".</p>
<p>Tobias Kuhn (<a href="http://orcid.org/0000-0002-1267-0234">http://orcid.org/0000-0002-1267-0234</a>)</p>
</div></div></div>  </div>

  
      <nav class="clearfix"><ul class="links inline"><li class="comment_forbidden first last"><span><a href="../user/login%3Fdestination=comment%252Freply%252F659%23comment-form.html">Log in</a> to post comments</span></li></ul></nav>
  
</article>

<div class="indented"><a id="comment-101"></a>
<article class="comment comment-by-node-author even last clearfix" about="/comment/101#comment-101" typeof="sioc:Post sioct:Comment">

  
  
      <header class="comment-header">
 
          <h3 property="dc:title" datatype="" class="comment-title"><a href="101.html#comment-101" rel="bookmark">Action letter</a></h3>
    
        
    <p class="comment-submitted"><span property="dc:date dc:created" content="2020-11-17T06:45:08-05:00" datatype="xsd:dateTime" rel="sioc:has_creator">Submitted by <span class="username" xml:lang="" about="/users/caspar-van-lissa" typeof="sioc:UserAccount" property="foaf:name" datatype="">Caspar Van Lissa</span> on <time datetime="2020-11-17T06:45:08Z"><span class="date-time">Tue, 11/17/2020 - 06:45</span></time></span><p>
  </header>
    
  <div class="comment-content">
    <span rel="sioc:reply_of" resource="/paper/worcs-workflow-open-reproducible-code-science" class="rdf-meta element-hidden"></span><span rel="sioc:reply_of" resource="/comment/100#comment-100" class="rdf-meta element-hidden"></span><div class="field field-name-comment-body field-type-text-long field-label-hidden view-mode-full"><div class="field-items"><div class="field-item even" property="content:encoded"><h1>WORCS: A Workflow for Open Reproducible Code in Science</h1>
<p>Dear Dr. Kuhn,</p>
<p>We would like to thank you for providing us with the opportunity to revise our manuscript, “WORCS: A Workflow for Open Reproducible Code in Science”. We sincerely appreciate the time you and the Reviewers have taken to read the manuscript, and the extensive and thought-provoking comments you have raised. We hereby submit our revision of the manuscript for your consideration. We have attempted to address each of the issues raised by yourself and the Reviewers on a point-by-point basis, and detail the changes made in the action letter below. We would like to thank you and the Reviewers for your time, and we look forward to receiving your response.</p>
<p>Sincerely, on behalve of all coauthors, Caspar van Lissa</p>
<h1>Reviewer #1</h1>
<p>Review #1 submitted on 19/Oct/2020 by Daniel Garijo ORCID logo <a href="https://orcid.org/0000-0003-0454-7145">https://orcid.org/0000-0003-0454-7145</a></p>
<ul><li>Overall Impression: Weak</li>
<li>Suggested Decision: Reject</li>
<li>Technical Quality of the paper: Good</li>
<li>Presentation: Good</li>
<li>Reviewer’s confidence: High</li>
<li>Significance: High significance</li>
<li>Background: Reasonable</li>
<li>Novelty: Limited novelty</li>
<li>Data availability: All used and produced data (if any) are FAIR and openly available in established data repositories</li>
<li>Length of the manuscript: The length of this manuscript is about right</li>
</ul><p>This paper describes worcs, a package and workflow to help make R code reproducible according to the TOP guidelines.</p>
<p>The paper describes a tutorial of the tool, and how it implements the TOP guidelines.</p>
<p>Reasons to accept:</p>
<ul><li>The paper is relevant to the journal and on a timely topic.</li>
<li>The capabilities of the tool for synthetic data generation, connect to OSF, preprint publishing and dependency gathering show a lot of promise.</li>
<li>All the software (except a 404 page in the help, which is minor) is available online; and the project seems maintained and well documented.</li>
</ul><p>Reasons to reject:</p>
<h2>C1 There is no user evaluation, and therefore all the claims about worcs are unsupported by evidence. If the authors had included a compelling evaluation I would be in favor of accepting the paper.</h2>
<p>Response: We appreciate the recommendation of performing a user evaluation, and this is certainly a next step in the development of WORCS. At this point, we have written this manuscript detailing the conceptual workflow, and released a software implementation for R-users. The next step, which we are now entering, is to build a user base, and then mount a user evaluation.</p>
<p>To address the Reviewer’s comment, we have set up an evaluation questionnaire, and sent it to the attendees of our previous workshops on WORCS. At the time of writing, responses are coming in. These are generally very favorable, and have not yet revealed major points of improvement.</p>
<p>We further want to point out that we have a strong history of addressing user comments submitted through the GitHub issues page. WORCS adheres to the Core Infrastructure Initiative “best practices” (CII Best Practices), which requires that the channels through which users can provide feedback are clearly indicated, as they are <a href="https://github.com/cjvanlissa/worcs#contributing-and-contact-information">here</a>. It is also evident from the <a href="https://github.com/cjvanlissa/worcs/issues?q=is%3Aissue+is%3Aclosed">resolved tickets on GitHub</a> that we have been very responsive in actively incorporating user feedback. This is another requirement of CII Best Practices.</p>
<p>We are committed to continue evaluating user experience, both actively through user evaluation forms, and passively through the GitHub issues pages. We hope that this commitment, along with the evidence of our prior responsiveness to user feedback presented above, help assuage the Reviewer’s concerns.</p>
<p>Finally, we wish to point out that the Editor has indicated that ‘resource papers do not need to have a full-blown user evaluation, but we expect “sound evidence of its (potential for) reuse”’. We therefore also elaborate on the efforts we have made to ensure that WORCS is reusable, followed by evidence of its current adoption rates.</p>
<p>We have made the following efforts to ensure potential for reuse:</p>
<ol><li>WORCS is implemented in an R-package, and thus readily available to R-users</li>
<li>This R-package is available in a standardized repository, namely CRAN (“The Comprehensive R Archive Network”), and can thus be installed through default channels</li>
<li>The software meets all criteria of the Core Infrastructure Initiative “best practices” (CII Best Practices), including clearly indicated channels for user feedback, and being responsive to that feedback</li>
<li>The R-package uses only tools that are fully integrated within the RStudio IDE, thus enhancing ease of use</li>
<li>The R-package is compatible with other reproducibility solutions, including Docker, drake, and make</li>
</ol><p>There is also evidence that WORCS is already being adopted, even prior to publication (as of 14-11-2020):</p>
<ol><li>The preprint has been downloaded 582 times, since being published on 31-05-2020</li>
<li>The <code>worcs</code> R-package has been downloaded 2567 times from CRAN, since being published on 18-05-2020</li>
<li>The GitHub project has been forked 8 times, watched 5 times, and starred 40 times</li>
<li>The lead author has given invited lectures on WORCS at:
<ul><li>Max-Planck-Institute for Human Development (2020)</li>
<li>Open Science Community Rotterdam (2020)</li>
<li>Eindhoven University of Technology (2020)</li>
</ul></li>
<li>WORCS is currently used in the following GitHub repositories (sorted by user):
<ul><li>cjvanlissa
<ul><li><a href="https://github.com/cjvanlissa/veni_sysrev">veni_sysrev</a></li>
<li><a href="https://github.com/cjvanlissa/IHBT-SEM">IHBT-SEM</a></li>
<li><a href="https://github.com/cjvanlissa/schumpe">schumpe</a></li>
<li><a href="https://github.com/cjvanlissa/storm_meta_father_mother">storm_meta_father_mother</a></li>
<li><a href="https://github.com/cjvanlissa/placebo_adhd">placebo_adhd</a></li>
<li><a href="https://github.com/cjvanlissa/mpib">mpib</a></li>
<li><a href="https://github.com/cjvanlissa/coping_covid">coping_covid</a></li>
</ul></li>
<li>IndrajeetPatil
<ul><li><a href="https://github.com/IndrajeetPatil/worcs_practice">worcs_practice</a></li>
</ul></li>
<li>EstherPlomp
<ul><li><a href="https://github.com/EstherPlomp/RDM-survey">RDM-survey</a></li>
</ul></li>
<li>lisallreiber
<ul><li><a href="https://github.com/lisallreiber/worcs_ws_empty">worcs_ws_empty</a></li>
</ul></li>
<li>KimBaldry
<ul><li><a href="https://github.com/KimBaldry/BIO-MATE">BIO-MATE</a></li>
</ul></li>
</ul></li>
</ol><p>Response: In addressing this comment, we respectfully refer to our response to the preceding comment (“There is no user evaluation”, R1 C1), and again emphasize the Editor’s comment that “resource papers do not need to have a full-blown user evaluation”. We would also like to clarify that the goal of the present manuscript is to define the problem WORCS intends to address (specifically; meeting the TOP-guidelines for Open Science), introduce the tools WORCS uses to address this problem, explain how WORCS uses these tools, and provide a step-by-step workflow. It is not clear what “claims” are made that are unsupported by evidence. In the revision, we have attempted to further clarify this goal from the paper onset:</p>
<p><strong>Abstract</strong>:</p>
<blockquote><p>This paper introduces the Workflow for Open Reproducible Code in Science (WORCS): A step-by-step procedure that researchers can follow to make a research project open and reproducible. The purpose of the workflow is to lower the threshold for adoption of open science principles. It is based on established best practices, and can be used either in parallel to, or in absence of, top-down requirements by journals, institutions, and funding bodies. To facilitate widespread adoption, the WORCS principles have been implemented in the R package <code>worcs</code>, which offers an RStudio project template and utility functions for specific workflow steps. This paper introduces the conceptual workflow, discusses how it meets different standards for open science, and addresses the functionality provided by the <code>R</code> implementation, <code>worcs</code>.</p>
</blockquote>
<p>We again emphasize that we take the recommendation to perform user evaluation very seriously, but we also feel that it would be premature to do so even before the paper is published. We have built WORCS based on established tools and best practices, and we have gathered preliminary positive feedback from the community through workshops and teaching.</p>
<h2>C3 Unsupported claims (beyond the ones derived from an absence of evaluation): The paper claims in many parts that worcs helps make an entire research project Open and Reproducible. However, at the same time there is also a claim for not supporting replication. Since replication is generally understood as a necessary steps towards reproducibility, I find this kind of contradictory. Maybe the claims can be modified stating that worcs helps documentation and understanding, hence supporting FAIR.</h2>
<p>Response: We thank the Reviewer for bringing this apparent contradiction to our attention. The misunderstanding arises from a lack of explicit definitions of the terms “replication” and “reproducibility” in the original manuscript. We are aware of the oft-lamented lack of consistent definitions for these terms (e.g., Patil, Peng, &amp; Leek, 2019; Plesser, 2018). In the revision, we have attempted to address this lack of clarity by including explicit definitions for these terms:</p>
<blockquote><p>Throughout this paper, we adhere to the definition of open science as “the practice of science in such a way that others can collaborate and contribute, where research data, lab notes and other research processes are freely available, under terms that enable reuse, redistribution and reproduction of the research and its underlying data and methods” (Vicente-Saez and Martinez-Fuentes 2018).<br />[…] reproducibility can be defined as “re-performing the same analysis with the same code and the same data using a different analyst”, and replicability as “re-performing the experiment and collecting new data” (Patil, Peng, and Leek 2019).</p>
</blockquote>
<p>After making these definitions explicit, it becomes clear that the Reviewer’s definition of replication as “a necessary steps towards reproducibility” is not consistent with the definitions used in the present paper.</p>
<p>We further wish to clarify that the fact that we address guidelines 1-7, but not guideline 8 (“replication”) is fully consistent with the intended purpose of the TOP-guidelines. Specifically, the first seven guidelines are intended to ensure the “reproducibility of the reported results based on the originating data” (direct quote from Nosek et al. 2015). The eighth criterion, by contrast, is “not formally a transparency standard for authors, [but] addresses journal guidelines for consideration of independent replications for publication” (again a direct quote). To avoid further confusion or controversy, we now address this distinction between the first 7 guidelines and the final 8th explicitly in the revised manuscript:</p>
<blockquote><p>These guidelines describe eight standards for open science, the first seven of which “account for reproducibility of the reported results based on the originating data, and for sharing sufficient information to conduct an independent replication” (Nosek et al. 2015). These guidelines are: 1) Comprehensive citation of literature, data, materials, and methods; 2) sharing data, 3) sharing the code required to reproduce analyses, 4) sharing new research materials, and 5) sharing details of the design and analysis; 6) pre-registration of studies before data collection, and 7) pre-registration of the analysis plan prior to analysis. The eighth criterion is “not formally a transparency standard for authors”, but “addresses journal guidelines for consideration of independent replications for publication”. In this context, reproducibility can be defined as “re-performing the same analysis with the same code using a different analyst”, and replicability as “re-performing the experiment and collecting new data” (Patil, Peng, and Leek 2019). WORCS defines the goals of open science in terms of the first seven TOP-guidelines, and the workflow and its <code>R</code> implementation are designed to facilitate compliance therewith.</p>
</blockquote>
<h2>C4: I am confused by the citation “Van Lissa, Caspar J., Aaron Peikert, and Andreas M. Brandmaier. 2020. “Worcs: Workflow for Open Reproducible Code in Science.”". Is this referring to another paper or the actual package? If it’s another paper then what is the contribution of this publication?</h2>
<p>Response: We apologize; it appears that the reference list was not parsed correctly after rendering the manuscript to HTML - the preferred format of Data Science. The reference is indeed to the R-package. In the revised manuscript, we have repaired the references so that it is clear that this is a software package.</p>
<h2>C5: If the paper is generated as DDG, where are the functions that show it’s capabilities? It is also not clear to me that some of the functionalities of Latex would be usable in RMarkdown, such as all the equations.</h2>
<p>Response: We understand that our introduction of the RMarkdown format was a bit too brief, and that readers may wonder whether specific functionalities of LaTeX would be usable in RMarkdown. To address this comment, the revised manuscript now reads:</p>
<blockquote><p>In the <code>R</code> implementation of WORCS, we recommend centering a research project around one dynamically generated <code>RMarkdown</code> document (Xie, Allaire, and Grolemund 2018). RMarkdown is based on <a href="https://en.wikipedia.org/wiki/Markdown"><code>Markdown</code></a>, which uses plain-text commands to typeset a document. It enhances this format with support for R-code, and is compatible with <a href="https://www.latex-project.org/">LaTeX</a> math typesetting. For a full overview of the functionality of <code>RMarkdown</code>, see Xie, Allaire, and Grolemund (2018).</p>
</blockquote>
<h2>C6: It is not clear how worcs addresses depedency management. The section lists a series of tools that can be used, but the section is not very specific to worcs.</h2>
<p>Response: We apologize for the apparent lack of clarity of this section. We have restructured and clarified this section to convey the following information:</p>
<ol><li>Introduce the general concept of dependency management</li>
<li>Introduce <code>renv</code>, the solution for dependency management used by the R-implementation of <code>worcs</code></li>
<li>Acknowledge the more comprehensive approach to dependency management offered by Docker</li>
</ol><p>The changes are extensive, so we will not quote them all here. Instead, we refer the Reviewer to the revised manuscript.</p>
<p>We have also, in response to another Reviewer comment, created a Docker Hub build that runs <code>worcs</code>. Therefore, as of this revision, both of the tools discussed in this section (i.e., <code>renv</code> and Docker) are directly relevant for WORCS-users, and the section should read less like a “series of tools that can be used”. If the Reviewer so desires, we would also be willing to cut the brief discussion of Code Ocean - but this had been included at the request of an informal reviewer of this paper (Daniël Lakens).</p>
<h2>C7a: The workflow of releasing data products on GitHub is not ideal. First, because datasets of 100 MBs usually require special handling. […] The authors seem to not describe very well what would happen in data science experiments where the data size is significant.</h2>
<p>Response: We have split this comment up to separately address the handling of large data, and findability (see below).</p>
<p>With regard to the handling of large data, we agree that GitHub is not an ideal platform for handling data larger than 100MB. It is important to note, however, that there are no restrictions for sharing files up to that size. Further, take into consideration that 100MB is very large for tabular data; for example, if only integer numbers are stored that is (100MB∗10242)/(22)=26214400(100MB∗10242)/(22)=26214400 unique values. In the social sciences, datasets rarely exceed this limit - and so for our readership, GitHub is as good a platform for sharing data as any. Moreover, GitHub is ideal for version controlling analysis code, RMarkdown documents, and small to medium-sized data files. In our opinion, the advantages of having <strong>all</strong> of a paper’s resources in the same repository outweighs the advantages of encouraging every reader to additionally use specialized solutions for “big data” (which most will not need).</p>
<p>To address this comment, we now explicitly address the problem of version controlling large files, and introduce a solution for larger files, which can be linked to git using <a href="https://git-lfs.github.com/">Git Large File Storage</a>:</p>
<blockquote><p>Data sharing may be impeded by several concerns. One technical concern is whether data require special storage provisions, as is often the case with “big data”. Most Git remote repositories do not support files larger than 100MB, but it is possible to enhance a Git repository with <a href="https://git-lfs.github.com/">Git Large File Storage</a> to add remotely stored files as large as several GB. Files exceeding this size require custom solutions beyond the scope of this paper.</p>
</blockquote>
<p>We also added a paragraph to the Limitations section that discusses the fact that no single workflow is suitable for all situations:</p>
<blockquote><p>Another limitation is the fact that no single workflow can suit all research projects. Not all projects conform to the steps outlined here; some may skip steps, add steps, or follow similar steps in a different order. In principle, WORCS does not enforce a linear order, and allows extentensive user flexibility. A related concern is that some projects might require specialized solutions outside of the scope of this paper. For example, some projects might use very large data files, rely on data that reside on a protected server, or use proprietary software dependencies that cannot be version controlled using <code>renv</code> or Docker. In such cases, the workflow can serve as a starting point for a custom approach; offering a primer on best practices, and a discussion of relevant factors to consider. We urge users to inform us of such challenges and custom solutions, so that we may add them to the <a href="https://github.com/cjvanlissa/worcs">overview of WORCS-projects on GitHub</a>, where they can serve as an example to others.</p>
</blockquote>
<h2>C7b: Second, it doesn’t follow very well the FAIR principles for findability and attribution of data. Instead, repositories like Zenodo or FigShare could be user to address data storage, with proper metadata and its corresponding citation.</h2>
<p>Response: We agree completely that it is important to make open data findable beyond just providing it freely. In order to address the Reviewer’s comment, we’ve revised the paper and the workflow vignette to emphasize more clearly that an essential step of the workflow is to connect the GitHub repository to a project page on the Open Science Framework - a standard repository for open research. We also devote more attention to the importance of DOIs; both for the project as a whole, and for specific resources, such as data files. Finally, we have elaborated on the discussion of the FAIR principles, emphasizing that WORCS is not designed to address these comprehensively, but pointing out specific ways in which WORCS is consistent with these guiding principles. Please see our response to Reviewer 2, comment 2 for more details.</p>
<p>With regard to the specific platforms named by the Reviewer: We are not aware of arguments to support a strong preference for Zenodo or FigShare over GitHub. Without such arguments, as far as we are concerned, these repositories are largely interchangeable. One key advantage of storing the data on GitHub is that they are then bundled with the documentation (e.g., codebook), analysis code, and manuscript. Thus, readers have access to <strong>all</strong> materials required for replication in one repository. Moreover, the entire project can be readily updated from the console, and is version controlled. GitHub additionally offers user interaction with the materials through forking, cloning, opening issues, and sending pull requests - all useful tools for replication and collaboration.</p>
<p>It is further worth noting that <a href="https://guides.github.com/activities/citable-code/">GitHub is compatible with Zenodo</a>; users can document GitHub repositories in Zenodo and generate a DOI. We now explicitly mention this option in the paper and workflow vignette. We also mention the potential redundancy with Steps 15-16 of the workflow, however: Connecting the GitHub repository to an OSF project and generating a DOI accomplishes effectively the same.</p>
<h2>C8: In the 3 phases listed, it is not clear what worcs does and does not for users.</h2>
<p>Response: It is not entirely clear to us whether the Reviewer is referring to the conceptual workflow (“WORCS”), or to the <code>R</code> implementation of WORCS, i.e., <code>worcs</code>.</p>
<p>We suspect that the Reviewer means the <code>worcs</code> R-package, so to address this comment, the Revision now states the following:</p>
<blockquote><p>We refer users of the <code>worcs</code> implementation in <code>R</code> to <a href="https://cjvanlissa.github.io/worcs/articles/setup.html">the workflow vignette</a>, which describes in more detail which steps are automated by the <code>R</code> package and which actions must be taken by the user. The vignette is also updated with future developments of the package.</p>
</blockquote>
<p>Just in case we misunderstood, and the Reviewer actually meant the conceptual workflow WORCS, we have also added a few sentences to clarify the purpose of the conceptual workflow:</p>
<blockquote><p>We provide a conceptual outline of the WORCS procedure below, based on WORCS Version 0.1.6. The conceptual workflow recommends actions to take in the different phases of a research project, in order to work openly and transparently in compliance with the TOP-guidelines and best practices for open science. Note that, although the steps are numbered for reference purposes, we acknowledge that the process of conducting research is not always linear.</p>
</blockquote>
<h2>C9: In some cases it looks like the authors claim that everyone should use R for open science. While R has a strong community behind, I believe there are others with very strong support (e.g. Python). I am not sure if suggesting a shift to R is the right approach. Note that this is the impression I got from reading the paper, maybe it was unintended by the authors.</h2>
<p>Response: We wish to emphasize that it was certainly unintended to suggest that everyone should us R for open science. To address this comment, we have done two things: First, throughout the paper, we have attempted to clarify the distinction between the platform independent conceptual workflow “WORCS”, and its implementation for R users, <code>worcs</code>. Second, throughout the paper, we encourage the implementation of WORCS in other languages. Most illustrative of these changes is the start of the paragraph on <strong>The R implementation of WORCS</strong>:</p>
<blockquote><p>WORCS is, first and foremost, a conceptual workflow that could be implemented in any software environment. As of this writing, the workflow has been implemented for <code>R</code> users in the package <code>worcs</code> (Van Lissa, Peikert, and Brandmaier 2020). Several arguments support our choice to implement this workflow first in <code>R</code> and RStudio, although we encourage developers to port the software to other languages.</p>
</blockquote>
<h2>C10: Vignete on citation leads to a 404.</h2>
<p>Response: We apologize for the oversight; we have fixed this URL and checked all URLs in the manuscript for mistakes (none found).</p>
<h2>C11: Synthetic data generation is fantastic, but it is not clear to me how does worcs actually keep the data consistent e.g. for training models. Does it follow a similar distribution to the source data?</h2>
<p>Response: We agree that synthetic data generation is exciting, but feel obliged to clarify that this is <strong>not</strong> a novel contribution of the present paper nor R-package. As indicated in the paper, we use the algorithm by Nowok and colleagues, who explain the details of the method. We re-implemented the method in the <code>worcs</code> package because the <code>synthpop</code> package is rarely maintained (e.g., <a href="https://github.com/bnowok/synthpop/pull/12">one of our pull requests</a> took almost 10 months to be accepted), and because we wanted a more flexible, customizable interface.</p>
<p>To answer the Reviewer’s specific question, the synthetic data indeed follow similar distributions to the source data, as indicated in the paper:</p>
<blockquote><p>Synthetic data mimic the level of measurement and (conditional) distributions of the real data (see Nowok, Raab, and Dibben 2016).</p>
</blockquote>
<p>We have additionally clarified the paragraph that introduces the <code>synthetic()</code> function, and added further references to the methodological paper, and to the documentation of the <code>synthetic()</code> function (which summarizes the algorithm).</p>
<h2>C12: Docker is deemed as complicated for novice users, but the setup vignete looks quite complex to me as well. Wouldn’t a Docker container with worcs installed be easier to use? Maybe these aspects would be better highlighted in a user evaluation.</h2>
<p>Response: We thank the Reviewer for this thoughtful suggestion. To address this comment, we have published an image on Docker Hub with all requirements of <code>worcs</code> installed, and added a Docker-specific setup vignette to the package and website, which is referenced in the Setup pagragraph:</p>
<blockquote><p>Before you can use the <code>R</code> implementation of the WORCS workflow, you have to install the required software. This 30 minute procedure is documented in <a href="https://cjvanlissa.github.io/worcs/articles/setup.html">the setup vignette</a>. After setting up your system, you can use <code>worcs</code> for all of your projects. Alternatively, Docker users can follow <a href="https://cjvanlissa.github.io/worcs/articles/setup-docker.html">the stetup with Docker vignette</a>.</p>
</blockquote>
<h1>Reviewer #2</h1>
<p>Submitted on 19/Oct/2020 by Remzi Celebi ORCID logo <a href="https://orcid.org/0000-0001-7769-4272">https://orcid.org/0000-0001-7769-4272</a></p>
<ul><li>Overall Impression: Good</li>
<li>Suggested Decision: Accept</li>
<li>Technical Quality of the paper: Good</li>
<li>Presentation: Good</li>
<li>Reviewer’s confidence: Medium</li>
<li>Significance: High significance</li>
<li>Background: Reasonable</li>
<li>Novelty: Limited novelty</li>
<li>Data availability: With exceptions that are admissible according to the data availability guidelines, all used and produced data are FAIR and openly available in established data repositories</li>
<li>Length of the manuscript: The length of this manuscript is about right</li>
</ul><p>The paper introduces the Workflow for Open Reproducible Code in Science (WORCS) , a workflow that complies with most of best practices for open sciences projects and demonstrates the use of the R worcs package for adapting WORCS workflow. The package provides a template for R users to create projects that follow open science principles.</p>
<p>Reasons to accept:</p>
<p>It provides a good discussion of TOP-guidelines (a set of open science principles) and how proposed workflow/package can be used to facilitate the adaptation of these guidelines.</p>
<p>Reasons to reject:</p>
<h2>C1: I would suggest the authors include a flowchart for users that shows the decision making process for which tools/operations should be used in WORCS. An example project would also help users better understand the package.</h2>
<p>Response: We thank the Reviewer for these suggestions. Regarding the flowchart - we had already included one in the previous version of the manuscript. To address this comment, we have clarified the references in the text to this flowchart:</p>
<blockquote><p>The WORCS workflow constitutes a step-by-step procedure that researchers can follow to make a research project open and reproducible. These steps are elaborated in greater detail below. See Figure 1 for a flowchart that highlights important steps in different stages of the research process.</p>
</blockquote>
<p>Regarding the example project, we have now curated a list of all public <code>worcs</code> projects on GitHub, and we reference this list in several places in the manuscript, including:</p>
<p><strong>Abstract</strong></p>
<blockquote><p>The source code for the <code>R</code> package and manuscript, and a list of <a href="https://github.com/cjvanlissa/worcs/tree/master#adoption-of-worcs-by-users">user examples of WORCS projects</a>, are available at <a href="https://github.com/cjvanlissa/worcs">https://github.com/cjvanlissa/worcs</a>.</p>
</blockquote>
<p><strong>Introducing the workflow</strong></p>
<blockquote><p>For examples of how the workflow is used in practice, we maintain <a href="https://github.com/cjvanlissa/worcs/tree/master#adoption-of-worcs-by-users">a list of public <code>worcs</code> projects</a> on the WORCS GitHub page.</p>
</blockquote>
<h2>C2: The claim that WORCS is amenable to the FAIR Guiding Principles should be justified. What are the solutions provided by WORCS that make a project interoperable and reusable? It would be improved with a more thorough discussion on how WORCS meets each FAIR principle.</h2>
<p>Response: To address this comment, we have made three changes:</p>
<ol><li>We have changed the workflow, emphasizing the importance of creating a project on the OSF and connecting it to the GitHub repository, and emphasizing the importance of DOIs and a license</li>
<li>We have rewritten the paragraph addressing the FAIR principles, with a more thorough discussion of how WORCS can help address each FAIR principle. We also state explicitly that <em>WORCS was not designed to comprehensively address these principles, but the workflow does facilitate meeting them.</em></li>
<li>We have made minor changes to the <code>worcs</code> package, e.g., to include information on the project readme about how readers can obtain access to data.</li>
</ol><p>This paragraph now reads:</p>
<blockquote><p>Aside from the TOP Guidelines, the FAIR Guiding Principles for scientific data management and stewardship (Wilkinson et al. 2016) are an important standard for open science. These principles advocate that digital research objects should be Findable, Accessible, Interoperable and Reusable. Initially mainly promoted as principles for data, they are increasingly applied as a standard for other types of research output as well, most notably software (Lamprecht et al. 2019).<br />WORCS was not designed to comprehensively address these principles, but the workflow does facilitate meeting them. In terms of Findability, all files in GitHub projects are searchable by content and meta-data, both on the platform and through standard search engines. The workflow further recommends that users create a project page on the Open Science Framework – a recognized repository in the social sciences. When doing so, it is important to generate a Digital Object Identifier (DOI) for this project. A DOI is a persistent way to identify and connect to an object on the internet, which is crucial for findability. Additional DOIs can be generated through the OSF for specific resources, such as data sets. It is further worth noting that, optionally, <a href="https://guides.github.com/activities/citable-code/">GitHub repositories can be connected to Zenodo</a>: On Zenodo, users can store a snapshot of the repository, provide metadata, and generate a DOI for the project or specific resources. When a project is connected to the OSF, as recommended, these steps may be redundant. Finally, each worcs project has a [YAML file] (<a href="https://yaml.org/">https://yaml.org/</a>) that lists the data objects (and other meta-data) in the project, which will allow web crawlers to index worcs projects. We use this metadata, for example, to identify public <code>worcs</code> projects on GitHub.<br />Accessibility is primarily safeguarded by hosting all project files in a public GitHub repository, which can be downloaded as a ZIP archive, and cloned or forked using the Git protocol. <a href="https://archiveprogram.github.com/">GitHub is committed to long term accessibility of these resources</a>. We further encourage users to make their data accessible when possible. When data must be closed, the <code>worcs</code> function <code>closed_data()</code> adds a paragraph to the project readme, indicating how readers can contact the author for access.<br />With regard to interoperability, the workflow recommends using plain-text files whenever possible, which ensures that code and meta-data are human- and machine-readable. Furthermore, the <code>worcs</code> package is implemented in R, which is open source. To promote reusability, we recommend users to specify an appropriate license for their projects. The <code>worcs</code> package includes the CC licenses, and suggests a CC-BY 4.0 license by default. This license is featured in the GitHub repository, and we encourage users to also display it on their project’s OSF page. To further facilitate reusability, <code>worcs</code> automatically generates a codebook for each data file. We encourage users to elaborate on the default codebook by adding variable description and categories, which would - in the future - enable indexing data by topic area.</p>
</blockquote>
<h2>C3: Is WORCS providing any service for de-identification?</h2>
<p>Response: WORCS does not provide a service for de-identification, and we consider this to be outside of the scope of the present paper. If the Reviewer feels differently, we are open to being convinced, however!</p>
<p>The current team lacks relevant expertise to make this contribution, but since <code>worcs</code> is open source software, anybody is welcome to contribute this functionality. Such a contribution would warrant coauthorship on the package, in line with the contribution guidelines <a href="https://github.com/cjvanlissa/worcs#contributing-and-contact-information">outlined here</a>.</p>
<h1>Reviewer #3</h1>
<p>Submitted on 26/Oct/2020 by Anonymous</p>
<ul><li>Overall Impression: Weak</li>
<li>Suggested Decision: Reject</li>
<li>Technical Quality of the paper: Average</li>
<li>Presentation: Good</li>
<li>Reviewer’s confidence: High</li>
<li>Significance: High significance</li>
<li>Background: Reasonable</li>
<li>Novelty: Limited novelty</li>
<li>Data availability: All used and produced data (if any) are FAIR and openly available in established data repositories</li>
<li>Length of the manuscript: The length of this manuscript is about right</li>
</ul><p>The paper presents WORCS, a library that is implemented within R-studio with the purpose of promoting the creation of reproducible research. In particular, they assist the researchers in the tasks of generating the scholarly article, thereby managing a number of tasks that researchers have to deal with from the reformulation of their hypothesis to version control, to dependency management, and finally generating the scholarly article.</p>
<p>Reasons to accept:</p>
<p>On the positive side, the authors built their tool, with Open Science principles in mind. In doing so, they have pined the actions that need to be performed as part of the research process. For instance, steps such as the pre-registration of the objective of the study before starting the investigation is ignored by existing solutions.</p>
<p>Reasons to reject:</p>
<p>The objective of the author is ambiguous. That said, I do not think that WORCS provide sufficient features that make it stands out from state of the art tools, for the following reasons.</p>
<h2>C1: The process as described by the author for conducting research is assumed to be linear. However, researchers often go back and forth on steps that they have already completed, and sometimes have to undo them completely and start from scratch. This does not seem to be tackled by the tool.</h2>
<p>Response: We agree with the Reviewer that researchers often go back and forth in their work. We did not intend to convey the impression that the process of conducting research is assumed to be linear. Of course, since we are presenting a workflow - there are a number of steps involved, which we have placed in roughly chronological order and numbered for reference purposes. For example, Study Design must logically occur prior to Writing and Analysis, and Publication can only logically occur after Writing. We also wish to emphasize that the “tool” (we assume the Reviewer is referring to the <code>worcs</code> R-package) does not enforce any kind of linearity.</p>
<p>To address the Reviewer’s comment, we have made two changes: First, we now explicitly state, in two places in the paper, and in the Workflow vignette included with the <code>worcs</code> package, that the steps are not necessarily linear:</p>
<p><strong>Introducing the workflow:</strong></p>
<blockquote><p>Note that, although the steps are numbered for reference purposes, we acknowledge that the process of conducting research is not always linear.</p>
</blockquote>
<p><strong>Limitations:</strong></p>
<blockquote><p>Another limitation is the fact that no single workflow can suit all research projects. Not all projects conform to the steps outlined here; some may skip steps, add steps, or follow similar steps in a different order. In principle, WORCS does not enforce a linear order, and allows extentensive user flexibility.</p>
</blockquote>
<p>Second, while it was always possible to manually add a manuscript or preregistration at a later point in time after creating a new project in RStudio - we have added more functions to increase the non-linear flexibility of the <code>worcs</code> package: <code>add_manuscript()</code> and <code>add_preregistration()</code>.</p>
<h2>C2: The gist of the tool seems to allow researchers to document their steps and generate a scholarly document that is ready for submission. In a sense, there is a similarity with existing notebooks such as Jupyter. So I was wondering why don’t the authors start from an existing popular tool such as Jupyter as the starting point (given the similarity in features that they would like to add), and extend Jupyter for instance with the new features such as dependency documentation and version control ?</h2>
<p>Response: There are two points we wish to address here. First, the Reviewer focuses on “the tool” - which we assume refers to the R package. However, the main goal of this paper is to introduce the conceptual workflow. The R package is merely offered to make adoption of the conceptual workflow easier for R users. This distinction is clarified several places throughout the manuscript, and to prevent further confusion, we have clarified the distinction even further in the revision, e.g.:</p>
<p>Abstract:</p>
<blockquote><p>This paper introduces the Workflow for Open Reproducible Code in Science (WORCS): A step-by-step procedure that researchers can follow to make a research project open and reproducible.<br />[…]<br />To facilitate widespread adoption, the WORCS principles have been implemented in the R package <code>worcs</code>, which offers an RStudio project template and utility functions for specific workflow steps.</p>
</blockquote>
<p>Manuscript, introduction:</p>
<blockquote><p>This paper introduces the conceptual workflow (referred to in upper case, “WORCS”), and discusses its underlying principles and how it meets different standards for open science. The principles underlying this workflow are universal and largely platform independent. We also present a software implementation of WORCS for <code>R</code> users (R Core Team 2020): The <code>R</code> package <code>worcs</code> (referred to in monospace font, Van Lissa, Peikert, and Brandmaier 2020).</p>
</blockquote>
<p>Manuscript, paragraph titled <strong>The R implementation of WORCS</strong>:</p>
<blockquote><p>WORCS is, first and foremost, a conceptual workflow that could be implemented in any software environment. As of this writing, the workflow has been implemented for <code>R</code> users in the package <code>worcs</code> (Van Lissa, Peikert, and Brandmaier 2020).</p>
</blockquote>
<p>Second, in response to the question “why don’t the authors start from an existing popular tool such as Jupyter”, we completely agree with the Reviewer that the conceptual workflow as explained in the paper could be implemented in any software environment, including Jupyter. In fact, we emphasize this in several places throughout the manuscript, for instance the paragraph titled <strong>The R Implementation of WORCS</strong>:</p>
<blockquote><p>WORCS is, first and foremost, a conceptual workflow that could be implemented in any software environment. As of this writing, the workflow has been implemented for <code>R</code> users in the package <code>worcs</code> (Van Lissa, Peikert, and Brandmaier 2020). Several arguments support our choice to implement this workflow first in <code>R</code> and RStudio, although we encourage developers to port the software to other languages.</p>
</blockquote>
<p>And in <strong>Future developments</strong>:</p>
<blockquote><p>although the workflow is currently implemented only in <code>R</code>, it is conceptually relevant for users of other statistical programming languages, and we welcome efforts to implement WORCS in other platforms."</p>
</blockquote>
<p>To answer the Reviewer’s question directly: We implemented this workflow in R because we are R developers. But in the paper, we also provide some arguments that compel our preference for R and RStudio as a choice for statistical computing in open science:</p>
<blockquote><p>First, <code>R</code> and all of it extensions are free open source software, which make it a tool of choice for open science. Second, all tools required for an open science workflow are implemented in <code>R</code>, most of these tools are directly accessible through the RStudio user interface, and some of them are actively developed by the team behind RStudio. Third, unlike any competitor, RStudio is a Public Benefit Corporation, whose primary purpose is explicitly aligned with open science principles:<br /><em>RStudio’s primary purpose is to create free and open-source software for data science, scientific research, and technical communication. This allows anyone with access to a computer to participate freely in a global economy that rewards data literacy; enhances the production and consumption of knowledge; and facilitates collaboration and reproducible research in science, education and industry.</em><br />Fourth, <code>R</code> is the second-most cited statistical software package (Muenchen 2012), following SPSS, which has no support for any of the open science tools discussed in this paper. Fifth, <code>R</code> is well-supported by a vibrant and inclusive online community, which develops new methods and packages, and provides support and tutorials for existing ones. Finally, <code>R</code> is highly interoperable: Packages are available to read and write nearly every filetype, including DOCX, PDF (in APA style), and HTML. Moreover, wrappers are available for many tools developed in other programming languages, and code written in other programming languages can be evaluated from <code>R</code>, including <code>C++</code>, <code>Fortran</code>, and <code>Python</code> (Allaire et al. 2020). There are excellent free resources for learning <code>R</code> (e.g., Grolemund and Wickham 2017).</p>
</blockquote>
<h2>C3: Regarding Version control and dependency documentation, the authors uses in their system existing solution, namely git and env, respectively. While I did not criticize reusing existing solution, on the contrary, I don’t believe that simply using such solution add a substantial value. For example, git capture changes at the level of the line of code/text, whereas a researcher needs an abstraction that informs him/her on the coarse-grained elements that are significant from the point of view of development and research steps so as to have a clear idea of where s/he is within the process and help him/her make the decision of the next steps to undertake.</h2>
<p>Response: Respectfully, we are not entirely sure what the criticism is here. First, as addressed in our response to the previous comment, there seems to be a misunderstanding about the scope of the paper. To clarify, we again state that our paper introduces a workflow for open science, based on objective criteria set out in the TOP-guidelines. Regarding novelty: Aside from the work-in-press of our co-authors - which is focused on strict computational reproducibility and is compatible with the present workflow - this is one of the first efforts to translate the requirements of open science into a step-by-step workflow. When you search Google Scholar for “open science workflow”, the first hit that actually describes a workflow is the present manuscript.</p>
<p>Second, the Reviewer claims that using e.g. Git does not add substantial value - but this claim is in disagreement with published literature (which we cite) that makes a strong case for the value of using Git for research (see Ram 2013; Blischak, Davenport, and Wilson 2016).</p>
<p>Third, there seems to be some misunderstanding of the workings of existing solutions. For example, the Reviewer remarks that “git capture changes at the level of the line of code/text, whereas a researcher needs an abstraction that informs him/her on the coarse-grained elements […] so as to have a clear idea of where s/he is within the process and help him/her make the decision of the next steps to undertake”. This comment suggests a fundamental misunderstanding of how Git works. Although Git tracks file changes on a line-by-line basis, such changes are typically grouped together in a “commit”, which is how Git captures the more “course-grained elements” the Reviewer refers to. For example, in our response to these reviews, we have created one Git commit for each Reviewer comment, and these commits contain changes to the action letter, manuscript, and vignettes.</p>
<p>To address the Reviewer’s comment, we now clarify this point in the revised manuscript:</p>
<blockquote><p>Git tracks changes to files on a line-by-line basis. The user can store these changes by making a “commit”: a snapshot of the version controlled files. Each “commit” can contain as many changes as desired; for example, a whole new paragraph, or many small changes made to address a single reviewer comment. Each commit is further tagged with a message, describing the changes. The <code>R</code> implementation of <code>worcs</code> contains a convenience function, <code>git_update("your commit message")</code>, which creates a new commit for all changes since the previous commit, labels it with “your commit message”, and pushes these changes to the remote repository. It is, effectively, a Git “quick save” command.</p>
</blockquote>
<p>The GitHub function of creating “releases” also helps mark the “course-grained” progress of a research project. In the WORCS procedure, we encourage taggin the preprint and submitted version of a manuscript as a “release” on GitHub. Such releases are prominently featured on the GitHub project page, and are accompanied by a downloadable archive of the historic state of the project. To further address the Reviewer’s comment, we now mention this in the manuscript text as well:</p>
<blockquote><p>The GitHub platform offers additional functionality over Git. One such feature is that specific stages in the lifecycle of a project can be <a href="https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/managing-releases-in-a-repository">tagged as a “release”</a>: A named downloadable snapshot of a specific state of the project, that is prominently featured on the GitHub project page. The WORCS procedure encourages users to create releases to demarcate project milestones such as preregistration and submission to a journal. Releases are also useful to mark the submission of a revised manuscript, and publication.</p>
</blockquote>
<h2>C4: To sum up, I think that the initial idea is interesting. However, I do not think that WORCS will be adopted by researchers or promote the state of the art of reproducibility, even if we focus on researchers who conduct their development and analysis using R.</h2>
<p>Response: We thank the Reviewer for acknowledging the idea to be interesting. Whether it will be adopted by researchers remains to be seen after publication. We can offer some prelimiary evidence that the workflow is already being adopted (as of 14-11-2020):</p>
<ol><li>The preprint has been downloaded 582 times, since being published on 31-05-2020</li>
<li>The <code>worcs</code> R-package has been downloaded 2567 times from CRAN, since being published on 18-05-2020</li>
<li>The GitHub project has been forked 8 times, watched 5 times, and starred 40 times</li>
<li>The lead author has given invited lectures on WORCS at:
<ul><li>Max-Planck-Institute for Human Development (2020)</li>
<li>Open Science Community Rotterdam (2020)</li>
<li>Eindhoven University of Technology (2020)</li>
</ul></li>
<li>WORCS is currently used in the following GitHub repositories (sorted by user):
<ul><li>cjvanlissa
<ul><li><a href="https://github.com/cjvanlissa/veni_sysrev">veni_sysrev</a></li>
<li><a href="https://github.com/cjvanlissa/IHBT-SEM">IHBT-SEM</a></li>
<li><a href="https://github.com/cjvanlissa/schumpe">schumpe</a></li>
<li><a href="https://github.com/cjvanlissa/storm_meta_father_mother">storm_meta_father_mother</a></li>
<li><a href="https://github.com/cjvanlissa/placebo_adhd">placebo_adhd</a></li>
<li><a href="https://github.com/cjvanlissa/mpib">mpib</a></li>
<li><a href="https://github.com/cjvanlissa/coping_covid">coping_covid</a></li>
</ul></li>
<li>IndrajeetPatil
<ul><li><a href="https://github.com/IndrajeetPatil/worcs_practice">worcs_practice</a></li>
</ul></li>
<li>EstherPlomp
<ul><li><a href="https://github.com/EstherPlomp/RDM-survey">RDM-survey</a></li>
</ul></li>
<li>lisallreiber
<ul><li><a href="https://github.com/lisallreiber/worcs_ws_empty">worcs_ws_empty</a></li>
</ul></li>
<li>KimBaldry
<ul><li><a href="https://github.com/KimBaldry/BIO-MATE">BIO-MATE</a></li>
</ul></li>
</ul></li>
</ol><h1>Reviewer #4</h1>
<p>Submitted on 26/Oct/2020 by Katherine Wolstencroft <a href="https://orcid.org/0000-0002-1279-5133">https://orcid.org/0000-0002-1279-5133</a></p>
<ul><li>Overall Impression: Good</li>
<li>Suggested Decision: Accept</li>
<li>Technical Quality of the paper: Excellent</li>
<li>Presentation: Good</li>
<li>Reviewer’s confidence: Medium</li>
<li>Significance: Moderate significance</li>
<li>Background: Reasonable</li>
<li>Novelty: Limited novelty</li>
<li>Data availability: All used and produced data (if any) are FAIR and openly available in established data repositories</li>
<li>Length of the manuscript: The length of this manuscript is about right</li>
</ul><p>This paper describes a Workflow for Open Reproducible Code in Science (WORCS). It promotes reproducible and FAIR R code and complies with the TOP-guidelines. The workflow is presented with an R library and a Github repository containing worcs templates and all supporting materials for this manuscript.</p>
<p>Reasons to accept:</p>
<p>The paper is well written and clearly describes WORCS, which promotes reproducible and open code for scientific investigations. It complies with current best practices for open and reproducible code and it also follows the FAIR principles. Such initiatives help promote better open and reproducible science. The approach taken here is pragmatic and lightweight, which is necessary for large scale uptake. The authors also focus on ease of use and provide a “one-click solution”.</p>
<p>WORCS is for scientists working with R with RMarkdown. It could therefore be argued that the utility is limited. However, R is widely used across science and the approach taken here could serve as an example for other software.</p>
<p>Reasons to reject:</p>
<h2>C1: The abstract of the paper states that the manuscript “provides examples of the implementation of worcs in R”, so I was expecting science projects that had been described using worcs. If these examples exist, they should be referenced and linked to from the manuscript, as they would provide a better demonstration of the practical utility of the workflow and software than tutorial style examples.</h2>
<p>Response: We thank the Reviewer for this helpful suggestion! Although such user examples are not yet numerous, the lead author has six public WORCS repositories, and there are four public repositories by other users. In our opinion, it would be most useful to have a continuously cumulating list of example repositories, and link that list in the paper - rather than to publish a static list. Therefore, to address this comment, we have scripted a web scraper that searches GitHub for WORCS repositories (using the metadata tags created by the <code>worcs</code> package). We have embedded the scraper in the README.Rmd file on the worcs GitHub page, which is regularly updated. In the revised paper, we point readers to <a href="https://github.com/cjvanlissa/worcs/tree/master#adoption-of-worcs-by-users">a list of public <code>worcs</code> projects</a> in the Abstract, and when introducing the workflow.</p>
<h2>C2: There is a comparison in the paper to another R-based reproducible science solution (Peikert and Brandmaier (2019)). The authors discuss some of the differences between these approaches, but the comparison is limited. It is not clear to me how different these approaches actually are, apart from ease of use. If scientists are already familiar with R, is there a significant learning curve with either solution?</h2>
<p>We agree that this is an important point. We have attemptedto address it by rewriting the paragraph comparing these two solutions (as cited below). We hope that this is satisfactory. At this point in time, this is all we can say about the comparison of the two workflows, because the workflow of Peikert and Brandmaier is still in press, and the software implementation has not yet been fully developed.</p>
<blockquote><p>A different class of solutions instead focuses on the practical issue of <em>how</em> researchers can meet the requirements of open science. A notable example is the workflow for reproducible analyses developed by Peikert and Brandmaier (n.d.), which uses Docker to ensure strict computational reproducibility for even the most sophisticated analyses. There are some decisive differences with WORCS. First, concerning the scope of the workflow: WORCS is designed to address a unique issue not covered by other existing solutions, namely, to provide a workflow most conducive to satisfying TOP-guidelines 1-7 while adhering to the FAIR principles. Peikert and Brandmaier instead focus primarily on computational reproducibility, which is relevant for TOP-guidelines 2, 3, and 5. Second, with regard to ease of use, WORCS aims to bring down the learning curve by adopting sensible defaults for any decisions to be made. Peikert and Brandmaier, by contrast, designed their workflow to be very flexible and comprehensive, thus requiring substantially more background knowledge and technical sophistication from users. To sum up, WORCS builds upon the same general principles as Peikert and Brandmaier, and the two workflows are compatible. What sets WORCS apart is that it is more lightweight in terms of system and user requirements, thereby facilitating adoption, but still ensures computational reproducibility under <em>most circumstances</em>.</p>
</blockquote>
<h1>Editor</h1>
<p>Submitted by Tobias Kuhn on Tue, 10/27/2020 - 02:04, <a href="http://orcid.org/0000-0002-1267-0234">http://orcid.org/0000-0002-1267-0234</a></p>
<h2>The reviewers agree that the paper has merit, but some of them also point to major shortcomings, in particular the lack of discussion of user adoption and of handling non-linear research processes. Note that according to our guidelines, resource papers do not need to have a full-blown user evaluation, but we expect “sound evidence of its (potential for) reuse”.</h2>
<p>Response: We thank the Editor and all Reviewers for their helpful comments on this manuscript. We have attempted to address all of these points in the action letter above. With regard to the specific point reiterated by the Editor:</p>
<h3>User adoption</h3>
<p>We thank the Editor for pointing out that a full-blown user evaluation is not required for a resource paper. Of course, we take this Reviewer suggestion very seriously, and intend to conduct a user evaluation after the paper is published and users begin to adopt the workflow. Please refer to our response to Reviewer 1’s comment 1, where we discuss our efforts to ensure the <em>potential for reuse</em>, and provide preliminary evidence for actual reuse/uptake of the workflow.</p>
<h3>Non-linear research processes</h3>
<p>Although Reviewer 3 claims that WORCS assumes a linear research process, this is incorrect.</p>
<p>We agree with Reviewer 3 that researchers often go back and forth in their work. We did not intend to convey the impression that the process of conducting research is assumed to be linear. Of course, since we are presenting a workflow - there are a number of steps involved, which we have placed in roughly chronological order and numbered for reference purposes. For example, Study Design must logically occur prior to Writing and Analysis, and Publication can only logically occur after Writing. We also wish to emphasize that the “tool” (we assume the Reviewer is referring to the <code>worcs</code> R-package) does not enforce any kind of linearity.</p>
<p>To address Reviewer 3’s comment, we have made two changes: First, we now explicitly state, in two places in the paper, and in the Workflow vignette included with the <code>worcs</code> package, that the steps are not necessarily linear:</p>
<p><strong>Introducing the workflow:</strong></p>
<blockquote><p>Note that, although the steps are numbered for reference purposes, we acknowledge that the process of conducting research is not always linear.</p>
</blockquote>
<p><strong>Limitations:</strong></p>
<blockquote><p>Another limitation is the fact that no single workflow can suit all research projects. Not all projects conform to the steps outlined here; some may skip steps, add steps, or follow similar steps in a different order. In principle, WORCS does not enforce a linear order, and allows extentensive user flexibility.</p>
</blockquote>
<p>Second, while it was always possible to manually add a manuscript or preregistration at a later point in time after creating a new project in RStudio - we have added more functions to increase the non-linear flexibility of the <code>worcs</code> package: <code>add_manuscript()</code> and <code>add_preregistration()</code>.</p>
<h1>References</h1>
<p>Allaire, J. J., Kevin Ushey, RStudio, and Yuan Tang. 2020. “R Markdown Python Engine.” 2020. <a href="https://rstudio.github.io/reticulate/articles/r_markdown.html">https://rstudio.github.io/reticulate/articles/r_markdown.html</a>.</p>
<p>Blischak, John D., Emily R. Davenport, and Greg Wilson. 2016. “A Quick Introduction to Version Control with Git and GitHub.” <em>PLOS Computational Biology</em> 12 (1): e1004668. <a href="https://doi.org/10.1371/journal.pcbi.1004668">https://doi.org/10.1371/journal.pcbi.1004668</a>.</p>
<p>Grolemund, Garrett, and Hadley Wickham. 2017. <em>R for Data Science</em>. O’Reilly. <a href="https://r4ds.had.co.nz/">https://r4ds.had.co.nz/</a>.</p>
<p>Lamprecht, Anna-Lena, Leyla Garcia, Mateusz Kuzak, Carlos Martinez, Ricardo Arcila, Eva Martin Del Pico, Victoria Dominguez Del Angel, et al. 2019. “Towards FAIR Principles for Research Software.” Edited by Paul Groth. <em>Data Science</em>, November, 1–23. <a href="https://doi.org/10.3233/DS-190026">https://doi.org/10.3233/DS-190026</a>.</p>
<p>Muenchen, Robert A. 2012. “The Popularity of Data Science Software.” R4stats.com. April 25, 2012. <a href="http://r4stats.com/articles/popularity/">http://r4stats.com/articles/popularity/</a>.</p>
<p>Nosek, B. A., G. Alter, G. C. Banks, D. Borsboom, S. D. Bowman, S. J. Breckler, S. Buck, et al. 2015. “Promoting an Open Research Culture.” <em>Science</em> 348 (6242): 1422–5. <a href="https://doi.org/10.1126/science.aab2374">https://doi.org/10.1126/science.aab2374</a>.</p>
<p>Nowok, Beata, Gillian M. Raab, and Chris Dibben. 2016. “Synthpop: Bespoke Creation of Synthetic Data in R.” <em>Journal of Statistical Software</em> 74 (1, 1): 1–26. <a href="https://doi.org/10.18637/jss.v074.i11">https://doi.org/10.18637/jss.v074.i11</a>.</p>
<p>Patil, Prasad, Roger D. Peng, and Jeffrey T. Leek. 2019. “A Visual Tool for Defining Reproducibility and Replicability.” <em>Nature Human Behaviour</em> 3 (7, 7): 650–52. <a href="https://doi.org/10.1038/s41562-019-0629-z">https://doi.org/10.1038/s41562-019-0629-z</a>.</p>
<p>Peikert, Aaron, and Andreas Markus Brandmaier. n.d. “A Reproducible Data Analysis Workflow with R Markdown, Git, Make, and Docker.” Accessed January 9, 2020. <a href="https://doi.org/10.31234/osf.io/8xzqy">https://doi.org/10.31234/osf.io/8xzqy</a>.</p>
<p>Ram, Karthik. 2013. “Git Can Facilitate Greater Reproducibility and Increased Transparency in Science.” <em>Source Code for Biology and Medicine</em> 8 (1): 7. <a href="https://doi.org/10.1186/1751-0473-8-7">https://doi.org/10.1186/1751-0473-8-7</a>.</p>
<p>R Core Team. 2020. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.</p>
<p>Van Lissa, Caspar J., Aaron Peikert, and Andreas M. Brandmaier. 2020. <em>Worcs: Workflow for Open Reproducible Code in Science</em> (version 0.1.5). <a href="https://cran.r-project.org/web/packages/worcs/index.html">https://cran.r-project.org/web/packages/worcs/index.html</a>.</p>
<p>Vicente-Saez, Ruben, and Clara Martinez-Fuentes. 2018. “Open Science Now: A Systematic Literature Review for an Integrated Definition.” <em>Journal of Business Research</em> 88 (July): 428–36. <a href="https://doi.org/10.1016/j.jbusres.2017.12.043">https://doi.org/10.1016/j.jbusres.2017.12.043</a>.</p>
<p>Wilkinson, Mark D., Michel Dumontier, IJsbrand Jan Aalbersberg, Gabrielle Appleton, Myles Axton, Arie Baak, Niklas Blomberg, et al. 2016. “The FAIR Guiding Principles for Scientific Data Management and Stewardship.” <em>Scientific Data</em> 3 (1): 160018. <a href="https://doi.org/10.1038/sdata.2016.18">https://doi.org/10.1038/sdata.2016.18</a>.</p>
<p>Xie, Yihui, J. J. Allaire, and Garrett Grolemund. 2018. <em>R Markdown: The Definitive Guide</em>. The R Series. Chapman and Hall/CRC. <a href="https://bookdown.org/yihui/rmarkdown/">https://bookdown.org/yihui/rmarkdown/</a>.</p>
</div></div></div>  </div>

  
      <nav class="clearfix"><ul class="links inline"><li class="comment_forbidden first last"><span><a href="../user/login%3Fdestination=comment%252Freply%252F659%23comment-form.html">Log in</a> to post comments</span></li></ul></nav>
  
</article>
</div>
  
</section>

  </div>
  </article>

  </div></div>                      </div>

                      
                    </div>
                  </div>

                </section>

                
              </div>
            </div>

                        
          </div>
        </div>

      </div>
    </div>

    
    
          <div id="footer-wrapper">
        <div class="container clearfix">
          <footer class="clearfix" role="contentinfo">
            <div class="region region-footer"><div class="region-inner clearfix"><div id="block-block-16" class="block block-block no-title odd first last block-count-4 block-region-footer block-16" ><div class="block-inner clearfix">  
  
  <div class="block-content content"><p style="text-align:center;"><a href="https://www.iospress.nl/disclaimer/" target="_blank">Disclaimer</a> | <a href="https://content.iospress.com/page/privacy" target="_blank">Privacy Policy</a> | Data Science is Published by <a href="http://www.iospress.com" target="_blank">IOS Press</a>, Copyright 2023</p>
</div>
  </div></div></div></div>            <p class="attribute-creator"></p>
          </footer>
        </div>
      </div>
    
  </div>
</div>
  </body>
</html>
